From a1ed20d918d9d124909d179558a95f2300e1b053 Mon Sep 17 00:00:00 2001
From: Mattia Rizzolo <mattia@debian.org>
Date: Mon, 6 Apr 2020 15:48:04 +0200
Subject: [PATCH] perltidy

Signed-off-by: Mattia Rizzolo <mattia@debian.org>
---
 scripts/checkbashisms.pl | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/scripts/checkbashisms.pl b/scripts/checkbashisms.pl
index fcb15fc3..653ec1bb 100755
--- a/scripts/checkbashisms.pl
+++ b/scripts/checkbashisms.pl
@@ -699,7 +699,8 @@ qr'(?:^|\s)(?<func>function\s)?\s*(?:[^<>\(\)\[\]\{\};|\s]*[^<>\(\)\[\]\{\};|\s\
         $LEADIN . qr'jobs\s'   => q<jobs>,
  #	$LEADIN . qr'jobs\s+-[^lp]\s' =>  q<'jobs' with option other than -l or -p>,
         $LEADIN
-          . qr'command\s+(?:-[pvV]+\s+)*-(?:[pvV])*[^pvV\s]' => q<'command' with option other than -p, -v or -V>,
+          . qr'command\s+(?:-[pvV]+\s+)*-(?:[pvV])*[^pvV\s]' =>
+          q<'command' with option other than -p, -v or -V>,
         $LEADIN
           . qr'setvar\s' =>
           q<setvar 'foo' 'bar' should be eval 'foo="'"$bar"'"'>,
-- 
2.25.0

