--- a/scripts/checkbashisms.pl	2020-01-30 21:48:57.000000000 +0100
+++ b/scripts/checkbashisms.pl	2020-03-18 10:29:47.216891685 +0100
@@ -712,9 +712,6 @@ qr'(?:^|\s)(?<func>function\s)?\s*(?:[^<
         $LEADIN
           . qr'(?:exit|return)\s+--' =>
           q<'exit --' should be 'exit' (idem for return)>,
-        $LEADIN
-          . qr'sleep\s+(?:-|\d+(?:[.a-z]|\s+\d))' =>
-          q<sleep only takes one integer>,
         $LEADIN . qr'hash(\s|\Z)' => q<hash>,
         qr'(?:[:=\s])~(?:[+-]|[+-]?\d+)(?:[/\s]|\Z)' =>
           q<non-standard tilde expansion>,
