diff -up a/plugins/aja/aja-routing.cpp b/aja-routing.cpp
--- a/plugins/aja/aja-routing.cpp	2023-06-19 00:05:22.000000000 +0200
+++ b/plugins/aja/aja-routing.cpp	2023-08-26 15:26:42.868582852 +0200
@@ -23,9 +23,9 @@ bool Routing::ParseRouteString(const std
 	blog(LOG_DEBUG, "aja::Routing::ParseRouteString: Input string: %s",
 	     route.c_str());
 
-	std::string route_lower(route);
-	route_lower = aja::lower(route_lower);
-	const std::string &route_strip = aja::replace(route_lower, " ", "");
+	std::string route_strip(route);
+	aja::lower(route_strip);
+	aja::replace(route_strip, " ", "");
 
 	if (route_strip.empty()) {
 		blog(LOG_DEBUG,
@@ -274,7 +274,7 @@ bool Routing::ConfigureSourceRoute(const
 		for (const auto &name : fs_associated) {
 			std::string placeholder = std::string(
 				name + "[{ch" + aja::to_string(c + 1) + "}]");
-			route_string = aja::replace(
+			aja::replace(
 				route_string, placeholder,
 				name + "[" +
 					aja::to_string(start_framestore_index) +
@@ -288,9 +288,8 @@ bool Routing::ConfigureSourceRoute(const
 	for (ULWord c = 0; c < NTV2_MAX_NUM_CHANNELS; c++) {
 		std::string channel_placeholder =
 			std::string("{ch" + aja::to_string(c + 1) + "}");
-		route_string =
-			aja::replace(route_string, channel_placeholder,
-				     aja::to_string(start_channel_index++));
+		aja::replace(route_string, channel_placeholder,
+			     aja::to_string(start_channel_index++));
 	}
 
 	if (!ParseRouteString(route_string, cnx))
@@ -421,7 +420,7 @@ bool Routing::ConfigureOutputRoute(const
 		for (const auto &name : fs_associated) {
 			std::string placeholder = std::string(
 				name + "[{ch" + aja::to_string(c + 1) + "}]");
-			route_string = aja::replace(
+			aja::replace(
 				route_string, placeholder,
 				name + "[" +
 					aja::to_string(start_framestore_index) +
@@ -435,9 +434,8 @@ bool Routing::ConfigureOutputRoute(const
 	for (ULWord c = 0; c < NTV2_MAX_NUM_CHANNELS; c++) {
 		std::string channel_placeholder =
 			std::string("{ch" + aja::to_string(c + 1) + "}");
-		route_string =
-			aja::replace(route_string, channel_placeholder,
-				     aja::to_string(start_channel_index++));
+		aja::replace(route_string, channel_placeholder,
+			     aja::to_string(start_channel_index++));
 	}
 
 	if (!ParseRouteString(route_string, cnx))
@@ -610,4 +608,4 @@ void Routing::LogRoutingPreset(const Rou
 	}
 }
 
-} // aja
+} // namespace aja
