From 0ac02fde22794d0b8219c32a0f1b02180bf36a92 Mon Sep 17 00:00:00 2001
From: Juan Jose Nicola <juan.nicola@greenbone.net>
Date: Tue, 22 Jan 2019 04:43:23 -0500
Subject: [PATCH] Remove calls to gvm_auth_tear_down() This function was
 removed from gvm-libs.

---
 src/gvmd.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/gvmd.c b/src/gvmd.c
index 66aacff63..fd375bb8d 100644
--- a/src/gvmd.c
+++ b/src/gvmd.c
@@ -906,9 +906,6 @@ cleanup ()
   g_debug ("   Exiting");
   if (log_config) log_config_free ();
 
-  /* Tear down authentication system conf, if any. */
-  gvm_auth_tear_down ();
-
   /* Delete pidfile if this process is the parent. */
   if (is_parent == 1) pidfile_remove ("gvmd");
 }
@@ -1254,7 +1251,6 @@ fork_update_nvt_cache ()
         cleanup_manage_process (FALSE);
         if (manager_socket > -1) close (manager_socket);
         if (manager_socket_2 > -1) close (manager_socket_2);
-        gvm_auth_tear_down ();
 
         /* Update the cache. */
 
