Description: Adjust syslog facility.
Author: unknown
Last-update: 2015-08-08
Index: arpalert-2.0.11/log.c
===================================================================
--- arpalert-2.0.11.orig/log.c
+++ arpalert-2.0.11/log.c
@@ -119,7 +119,7 @@ void logmsg(int priority, const char *fm
 	#ifdef USE_SYSLOG
 	if(config[CF_USESYSLOG].valeur.integer == TRUE &&
 	   syslog_initialized == TRUE){
-		syslog(priority, log_msg); 
+		syslog(priority, "%s", log_msg); 
 	}
 	#endif
 
