Description: spelling errors detected by lintian
Author: Thorsten Alteholz <debian@alteholz.de>
Index: arpalert-2.0.12/maclist.c
===================================================================
--- arpalert-2.0.12.orig/maclist.c	2020-08-06 20:18:10.000000000 +0000
+++ arpalert-2.0.12/maclist.c	2020-08-06 20:47:25.662316453 +0000
@@ -197,7 +197,7 @@
 				}
 				else {
 					logmsg(LOG_ERR,
-					       "file: \"%s\", line %d: flag \"%s\" not availaible",
+					       "file: \"%s\", line %d: flag \"%s\" not available",
 					       file_name, ligne, args[i]);
 					return -1;
 				}
