--- a/extract.c.orig	2010-03-02 03:49:30.000000000 +0100
+++ b/extract.c	2015-03-18 10:15:24.489219146 +0100
@@ -1868,7 +1868,12 @@ unsigned char *extract_jpeg(f_state *s,
 		if (((foundat + headersize) - buf) > buflen){ return NULL; }	
 
 		foundat += headersize;
-		
+
+		if (foundat >= (buf + buflen))
+			{
+			return buf + needle->header_len;
+			}
+
 		if (foundat[2] != (unsigned char)'\xff')
 			{
 			break;
