--- a/Makefile.orig	2010-03-02 03:49:30.000000000 +0100
+++ b/Makefile	2015-03-18 10:37:59.000000000 +0100
@@ -24,9 +24,9 @@ MAN_PAGES = $(NAME).8.gz
 RAW_FLAGS += -DVERSION=\"$(VERSION)\"
 
 # Where we get installed
-BIN = /usr/local/bin
-MAN = /usr/share/man/man8
-CONF= /usr/local/etc
+BIN = $(DESTDIR)/usr/bin
+MAN = $(DESTDIR)/usr/man/man8
+CONF= $(DESTDIR)/etc
 # Setup for compiling and cross-compiling for Windows
 # The CR_ prefix refers to cross compiling from OSX to Windows
 CR_CC = $(CR_BASE)/gcc
@@ -118,9 +118,9 @@ foremost: $(OBJ)
 #---------------------------------------------------------------------
 
 install: goals
-	install -m 755 $(NAME) $(BIN)
-	install -m 444 $(MAN_PAGES) $(MAN)
-	install -m 444 foremost.conf $(CONF)
+	install -D -m 755 $(NAME) $(BIN)/$(NAME)
+	install -D -m 644 $(MAN_PAGES) $(MAN)/$(MAN_PAGES)
+	install -D -m 644 foremost.conf $(CONF)/foremost.conf
 macinstall: BIN = /usr/local/bin/
 macinstall: MAN = /usr/share/man/man1/
 macinstall: CONF = /usr/local/etc/
