diff -Napur a/Makefile.am b/Makefile.am
--- a/Makefile.am	2019-03-03 22:17:18.000000000 +0100
+++ b/Makefile.am	2019-09-17 13:16:59.035608850 +0200
@@ -1,6 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4 -I .
 
-SUBDIRS = include lib bin etc doc man
+SUBDIRS = include lib bin etc man
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = varnishapi.pc
diff -Napur a/Makefile.in b/Makefile.in
--- a/Makefile.in	2019-09-16 11:47:21.000000000 +0200
+++ b/Makefile.in	2019-09-17 13:17:24.130608401 +0200
@@ -417,7 +417,7 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4 -I .
-SUBDIRS = include lib bin etc doc man
+SUBDIRS = include lib bin etc man
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = varnishapi.pc
 m4dir = $(datadir)/aclocal
diff -Napur a/configure.ac b/configure.ac
--- a/configure.ac	2019-09-16 11:46:39.000000000 +0200
+++ b/configure.ac	2019-09-17 13:18:44.935606954 +0200
@@ -48,17 +48,6 @@ if test "x$RST2MAN" = "xno"; then
     [rst2man is needed to build Varnish, please install python3-docutils.])
 fi
 
-AC_ARG_WITH([sphinx-build],
-  AS_HELP_STRING([--with-sphinx-build=PATH], [Location of sphinx-build (auto)]),
-  [SPHINX="$withval"],
-  AC_CHECK_PROGS(SPHINX,
-    [sphinx-build-3.6 sphinx-build-3 sphinx-build],
-    [no]))
-if test "x$SPHINX" = "xno"; then
-  AC_MSG_ERROR(
-    [sphinx-build is needed to build Varnish, please install python3-sphinx.])
-fi
-
 AC_ARG_WITH([rst2html],
   AS_HELP_STRING([--with-rst2html=PATH], [Location of rst2html (auto)]),
   [RST2HTML="$withval"],
@@ -804,9 +793,6 @@ VSC_GEN_H="$(echo $VSC_SRC | sed 's:\.vs
 AC_SUBST(VSC_GEN_C)
 AC_SUBST(VSC_GEN_H)
 
-# Make sure this include dir exists
-AC_CONFIG_COMMANDS([mkdir], [$MKDIR_P doc/sphinx/include])
-
 # Generate output
 AC_CONFIG_FILES([
     Makefile
@@ -819,10 +805,6 @@ AC_CONFIG_FILES([
     bin/varnishhist/Makefile
     bin/varnishtest/Makefile
     bin/varnishncsa/Makefile
-    doc/Makefile
-    doc/graphviz/Makefile
-    doc/sphinx/Makefile
-    doc/sphinx/conf.py
     etc/Makefile
     include/Makefile
     lib/Makefile
