Use apm@1.11.0 with Node v4 and apm v3 #11897 
diff -Naupr atom-1.8.0.orig/apm/package.json atom-1.8.0/apm/package.json
--- atom-1.8.0.orig/apm/package.json	2016-06-06 23:29:19.000000000 +0200
+++ atom-1.8.0/apm/package.json	2016-06-07 08:43:29.457123683 +0200
@@ -6,6 +6,6 @@
     "url": "https://github.com/atom/atom.git"
   },
   "dependencies": {
-    "atom-package-manager": "1.9.2"
+    "atom-package-manager": "1.11.1"
   }
 }
diff -Naupr atom-1.8.0.orig/build/package.json atom-1.8.0/build/package.json
--- atom-1.8.0.orig/build/package.json	2016-06-06 23:29:19.000000000 +0200
+++ atom-1.8.0/build/package.json	2016-06-07 08:43:15.191302031 +0200
@@ -32,7 +32,7 @@
     "grunt-standard": "^2.0.0",
     "legal-eagle": "~0.13.0",
     "minidump": "~0.9",
-    "npm": "2.13.3",
+    "npm": "3.9.6",
     "rcedit": "~0.3.0",
     "request": "~2.27.0",
     "rimraf": "~2.2.2",
diff -Naupr atom-1.8.0.orig/script/bootstrap atom-1.8.0/script/bootstrap
--- atom-1.8.0.orig/script/bootstrap	2016-06-06 23:29:19.000000000 +0200
+++ atom-1.8.0/script/bootstrap	2016-06-07 08:44:32.442336260 +0200
@@ -75,7 +75,7 @@ function bootstrap() {
 
   var buildInstallCommand = initialNpmCommand + npmFlags + 'install';
   var buildInstallOptions = {cwd: path.resolve(__dirname, '..', 'build')};
-  var apmInstallCommand = npmPath + npmFlags + '--target=0.10.35 ' + 'install';
+  var apmInstallCommand = npmPath + npmFlags + '--target=4.4.5 --global-style ' + 'install';
   var apmInstallOptions = {cwd: apmInstallPath};
   var moduleInstallCommand = apmPath + ' install' + apmFlags;
   var dedupeApmCommand = apmPath + ' dedupe' + apmFlags;
@@ -123,11 +123,6 @@ function bootstrap() {
       options: apmInstallOptions
     },
     {
-      command: apmPath + ' clean' + apmFlags,
-      message: 'Deleting old packages...',
-      options: moduleInstallOptions
-    },
-    {
       command: moduleInstallCommand,
       options: moduleInstallOptions
     },
