Update Electron to 0.36.12 #11348 
https://github.com/atom/atom/pull/11348
diff -Naupr atom-1.8.0.orig/package.json atom-1.8.0/package.json
--- atom-1.8.0.orig/package.json	2016-06-06 23:29:19.000000000 +0200
+++ atom-1.8.0/package.json	2016-06-07 07:44:25.649427376 +0200
@@ -12,7 +12,7 @@
     "url": "https://github.com/atom/atom/issues"
   },
   "license": "MIT",
-  "electronVersion": "0.36.8",
+  "electronVersion": "0.36.12",
   "dependencies": {
     "async": "0.2.6",
     "atom-keymap": "6.3.2",
diff -Naupr atom-1.8.0.orig/spec/atom-environment-spec.coffee atom-1.8.0/spec/atom-environment-spec.coffee
--- atom-1.8.0.orig/spec/atom-environment-spec.coffee	2016-06-06 23:29:19.000000000 +0200
+++ atom-1.8.0/spec/atom-environment-spec.coffee	2016-06-07 07:44:17.069534640 +0200
@@ -365,7 +365,7 @@ describe "AtomEnvironment", ->
       updateAvailableHandler = jasmine.createSpy("update-available-handler")
       subscription = atom.onUpdateAvailable updateAvailableHandler
 
-      autoUpdater = require('remote').require('auto-updater')
+      autoUpdater = require('electron').remote.require('auto-updater')
       autoUpdater.emit 'update-downloaded', null, "notes", "version"
 
       waitsFor ->
diff -Naupr atom-1.8.0.orig/spec/module-cache-spec.coffee atom-1.8.0/spec/module-cache-spec.coffee
--- atom-1.8.0.orig/spec/module-cache-spec.coffee	2016-06-06 23:29:19.000000000 +0200
+++ atom-1.8.0/spec/module-cache-spec.coffee	2016-06-07 07:43:49.374880871 +0200
@@ -8,13 +8,13 @@ describe 'ModuleCache', ->
   beforeEach ->
     spyOn(Module, '_findPath').andCallThrough()
 
-  it 'resolves atom shell module paths without hitting the filesystem', ->
+  it 'resolves Electron module paths without hitting the filesystem', ->
     builtins = ModuleCache.cache.builtins
     expect(Object.keys(builtins).length).toBeGreaterThan 0
 
     for builtinName, builtinPath of builtins
       expect(require.resolve(builtinName)).toBe builtinPath
-      expect(fs.isFileSync(require.resolve(builtinName)))
+      expect(fs.isFileSync(require.resolve(builtinName))).toBeTruthy()
 
     expect(Module._findPath.callCount).toBe 0
 
diff -Naupr atom-1.8.0.orig/spec/window-event-handler-spec.coffee atom-1.8.0/spec/window-event-handler-spec.coffee
--- atom-1.8.0.orig/spec/window-event-handler-spec.coffee	2016-06-06 23:29:19.000000000 +0200
+++ atom-1.8.0/spec/window-event-handler-spec.coffee	2016-06-07 07:43:02.754463706 +0200
@@ -75,7 +75,7 @@ describe "WindowEventHandler", ->
 
   describe "when a link is clicked", ->
     it "opens the http/https links in an external application", ->
-      shell = require 'shell'
+      shell = require 'electron'
       spyOn(shell, 'openExternal')
 
       link = document.createElement('a')
diff -Naupr atom-1.8.0.orig/src/module-cache.coffee atom-1.8.0/src/module-cache.coffee
--- atom-1.8.0.orig/src/module-cache.coffee	2016-06-06 23:29:19.000000000 +0200
+++ atom-1.8.0/src/module-cache.coffee	2016-06-07 07:42:34.771813537 +0200
@@ -208,7 +208,7 @@ registerBuiltins = (devMode) ->
     cache.builtins[builtin] = path.join(commonRoot, "#{builtin}.js")
 
   rendererRoot = path.join(atomShellRoot, 'renderer', 'api', 'lib')
-  rendererBuiltins = ['ipc-renderer', 'remote']
+  rendererBuiltins = ['ipc-renderer', 'remote', 'screen']
   for builtin in rendererBuiltins
     cache.builtins[builtin] = path.join(rendererRoot, "#{builtin}.js")
 
