--- Makefile.orig	2014-09-06 15:12:52.000000000 +0200
+++ Makefile	2014-10-09 07:54:13.336662075 +0200
@@ -49,6 +49,8 @@ install:	atop atopacctd
 			then mkdir -p $(DESTDIR)$(LOGPATH); fi
 		if [ ! -d $(DESTDIR)$(BINPATH) ]; 	\
 			then mkdir -p $(DESTDIR)$(BINPATH); fi
+		if [ ! -d $(DESTDIR)$(SBINPATH) ]; 	\
+			then mkdir -p $(DESTDIR)$(SBINPATH); fi
 		if [ ! -d $(DESTDIR)$(SCRPATH) ]; 	\
 			then mkdir -p $(DESTDIR)$(SCRPATH); fi
 		if [ ! -d $(DESTDIR)$(MAN1PATH) ]; 	\
@@ -81,15 +83,14 @@ install:	atop atopacctd
 		chmod 0700 		$(DESTDIR)$(SBINPATH)/atopacctd
 		cp atop   		$(DESTDIR)$(BINPATH)/atop-$(VERS)
 		ln -sf atop-$(VERS)     $(DESTDIR)$(BINPATH)/atopsar-$(VERS)
-		cp atop.daily    	$(DESTDIR)$(SCRPATH)
-		chmod 0711 	 	$(DESTDIR)$(SCRPATH)/atop.daily
+		cp atop.daily    	$(DESTDIR)$(CRNPATH)
+		chmod 0755 	 	$(DESTDIR)$(CRNPATH)/atop.daily
 		cp man/atop.1    	$(DESTDIR)$(MAN1PATH)
 		cp man/atopsar.1 	$(DESTDIR)$(MAN1PATH)
 		cp man/atoprc.5  	$(DESTDIR)$(MAN5PATH)
 		cp man/atopacctd.8  	$(DESTDIR)$(MAN8PATH)
 		cp atopacct.init     	$(DESTDIR)$(INIPATH)/atopacct
-		cp atop.init     	$(DESTDIR)$(INIPATH)/atop
-		cp atop.cron     	$(DESTDIR)$(CRNPATH)/atop
+		cp atop.init     	$(DESTDIR)$(INIPATH)/rc.atop
 		cp psaccs_atop   	$(DESTDIR)$(ROTPATH)/psaccs_atop
 		cp psaccu_atop  	$(DESTDIR)$(ROTPATH)/psaccu_atop
 		touch          	  	$(DESTDIR)$(LOGPATH)/dummy_before
