diff -Naupr a/Makefile b/Makefile
--- a/Makefile	2019-01-12 23:37:19.000000000 +0100
+++ b/Makefile	2019-01-13 12:36:51.363700046 +0100
@@ -91,7 +91,7 @@ sysvinstall:	genericinstall
 		if [ ! -d $(DESTDIR)$(INIPATH) ]; 			\
 		then	mkdir -p  $(DESTDIR)$(INIPATH); fi
 		#
-		cp atop.init      $(DESTDIR)$(INIPATH)/atop
+		cp atop.init      $(DESTDIR)$(INIPATH)/rc.atop
 		cp atopacct.init  $(DESTDIR)$(INIPATH)/atopacct
 		cp atop.cronsysv  $(DESTDIR)$(CRNPATH)/atop
 		#
@@ -158,8 +158,8 @@ genericinstall:	atop atopacctd atopconve
 		cp atopconvert 		$(DESTDIR)$(BINPATH)/atopconvert
 		chown root		$(DESTDIR)$(BINPATH)/atopconvert
 		chmod 0711 		$(DESTDIR)$(BINPATH)/atopconvert
-		cp atop.daily    	$(DESTDIR)$(SCRPATH)
-		chmod 0711 	 	$(DESTDIR)$(SCRPATH)/atop.daily
+		cp atop.daily    	$(DESTDIR)$(CRNPATH)
+		chmod 0755	 	$(DESTDIR)$(CRNPATH)/atop.daily
 		cp man/atop.1    	$(DESTDIR)$(MAN1PATH)
 		cp man/atopsar.1 	$(DESTDIR)$(MAN1PATH)
 		cp man/atopconvert.1 	$(DESTDIR)$(MAN1PATH)
