diff -Naupr a/Makefile b/Makefile
--- a/Makefile	2024-01-04 11:06:28.000000000 +0100
+++ b/Makefile	2024-01-09 11:51:56.861915210 +0100
@@ -3,24 +3,24 @@
 # Gerlof Langeveld - gerlof.langeveld@atoptool.nl
 #
 DESTDIR  =
+PKGFLAGS =
 
 BINPATH  = /usr/bin
 SBINPATH = /usr/sbin
-SCRPATH  = /usr/share/atop
 LOGPATH  = /var/log/atop
-MAN1PATH = /usr/share/man/man1
-MAN5PATH = /usr/share/man/man5
-MAN8PATH = /usr/share/man/man8
-INIPATH  = /etc/init.d
+MAN1PATH = /usr/man/man1
+MAN5PATH = /usr/man/man5
+MAN8PATH = /usr/man/man8
+INIPATH  = /etc/rc.d
 DEFPATH  = /etc/default
 SYSDPATH = /lib/systemd/system
-CRNPATH  = /etc/cron.d
+DAYPATH  = /etc/cron.daily
 ROTPATH  = /etc/logrotate.d
 PMPATH1  = /usr/lib/pm-utils/sleep.d
 PMPATH2  = /usr/lib64/pm-utils/sleep.d
 PMPATHD  = /usr/lib/systemd/system-sleep
 
-CFLAGS  += -O2 -I. -Wall $(shell pkg-config --cflags glib-2.0)  -Wno-stringop-truncation -Wmissing-prototypes -Wmissing-declarations -Wformat-security # -DNOPERFEVENT   # -DHTTPSTATS
+CFLAGS  += $(PKGFLAGS) -O2 -I. -Wall $(shell pkg-config --cflags glib-2.0)  -Wno-stringop-truncation -Wmissing-prototypes -Wmissing-declarations -Wformat-security # -DNOPERFEVENT   # -DHTTPSTATS
 LDFLAGS  = $(shell pkg-config --libs glib-2.0)
 OBJMOD0  = version.o
 OBJMOD1  = various.o  deviate.o   procdbase.o
@@ -96,18 +96,15 @@ install:	genericinstall
 sysvinstall:	genericinstall
 		if [ ! -d $(DESTDIR)$(INIPATH) ]; 			\
 		then	mkdir -p  $(DESTDIR)$(INIPATH);	fi
-		if [ ! -d $(DESTDIR)$(SCRPATH) ]; 			\
-		then	mkdir -p $(DESTDIR)$(SCRPATH);	fi
-		if [ ! -d $(DESTDIR)$(CRNPATH) ]; 			\
-		then	mkdir -p $(DESTDIR)$(CRNPATH);	fi
+		if [ ! -d $(DESTDIR)$(DAYPATH) ]; 			\
+		then	mkdir -p $(DESTDIR)$(DAYPATH);	fi
 		if [ ! -d $(DESTDIR)$(ROTPATH) ]; 			\
 		then	mkdir -p $(DESTDIR)$(ROTPATH);	fi
 		#
-		cp atop.init      $(DESTDIR)$(INIPATH)/atop
-		cp atopacct.init  $(DESTDIR)$(INIPATH)/atopacct
-		cp atop.cronsysv  $(DESTDIR)$(CRNPATH)/atop
-		cp atop.daily     $(DESTDIR)$(SCRPATH)
-		chmod 0711        $(DESTDIR)$(SCRPATH)/atop.daily
+		cp atop.init      $(DESTDIR)$(INIPATH)/rc.atop
+		cp atopacct.init  $(DESTDIR)$(INIPATH)/rc.atopacct
+		cp atop.daily     $(DESTDIR)$(DAYPATH)
+		chmod 0711        $(DESTDIR)$(DAYPATH)/atop.daily
 		touch             $(DESTDIR)$(LOGPATH)/dummy_before
 		touch             $(DESTDIR)$(LOGPATH)/dummy_after
 		#
