diff -purwbB adns-1.5.0.orig/Makefile.in adns-1.5.0/Makefile.in
--- adns-1.5.0.orig/Makefile.in	2014-11-02 16:59:59.000000000 +0200
+++ adns-1.5.0/Makefile.in	2017-10-23 21:39:58.017253534 +0300
@@ -75,10 +75,10 @@ adnshost.txt: all
 
 web-install: adnshost.txt
 	test -e $(WEBDIR) || mkdir $(WEBDIR)
-	$(INSTALL_DATA) $(srcdir)/README.html $(WEBDIR)/
-	$(INSTALL_DATA) $(srcdir)/src/adns.h $(WEBDIR)/adns.h.txt
-	$(INSTALL_DATA) COPYING $(WEBDIR)/COPYING.txt
-	$(INSTALL_DATA) adnshost.txt $(WEBDIR)/
+	$(INSTALL_DATA) $(srcdir)/README.html $(DESTDIR)$(WEBDIR)/
+	$(INSTALL_DATA) $(srcdir)/src/adns.h $(DESTDIR)$(WEBDIR)/adns.h.txt
+	$(INSTALL_DATA) COPYING $(DESTDIR)$(WEBDIR)/COPYING.txt
+	$(INSTALL_DATA) adnshost.txt $(DESTDIR)$(WEBDIR)/
 
 check:			all
 	$(MAKE) -C regress check
diff -purwbB adns-1.5.0.orig/client/Makefile.in adns-1.5.0/client/Makefile.in
--- adns-1.5.0.orig/client/Makefile.in	2014-10-20 02:07:01.000000000 +0300
+++ adns-1.5.0/client/Makefile.in	2017-10-23 21:50:15.172314163 +0300
@@ -58,9 +58,9 @@ ALL_OBJS=	$(ADH_OBJS) $(TARG_OBJS)
 all:		$(TARGETS)
 
 install:	$(TARG_INSTALL)
-		mkdir -p $(bindir)
+		$(INSTALL_DIR) $(DESTDIR)$(bindir)
 		set -xe; for f in $(TARG_INSTALL); \
-			do $(INSTALL_PROGRAM) $$f $(bindir)/$$f; done
+			do $(INSTALL_PROGRAM) $$f $(DESTDIR)$(bindir)/$$f; done
 
 uninstall:
 		for f in $(TARGETS); do rm -f $(bindir)/$$f; done
diff -purwbB adns-1.5.0.orig/common.make.in adns-1.5.0/common.make.in
--- adns-1.5.0.orig/common.make.in	2014-10-20 03:28:30.000000000 +0300
+++ adns-1.5.0/common.make.in	2017-10-23 21:51:32.964321805 +0300
@@ -60,4 +60,4 @@ endif
 
 INSTALL_PROGRAM=	$(INSTALL) -m 755 $(INSTALL_PROGRAM_FLAGS)
 INSTALL_DATA=		$(INSTALL) -m 644
-
+INSTALL_DIR=		$(INSTALL) -d 755
diff -purwbB adns-1.5.0.orig/dynamic/Makefile.in adns-1.5.0/dynamic/Makefile.in
--- adns-1.5.0.orig/dynamic/Makefile.in	2014-10-20 02:07:01.000000000 +0300
+++ adns-1.5.0/dynamic/Makefile.in	2017-10-23 21:39:11.713248985 +0300
@@ -31,9 +31,9 @@ ALLOBJS=	$(addsuffix _p.o, $(basename $(
 
 install:
 		mkdir -p $(libdir)
-		$(INSTALL_PROGRAM) $(SHLIBFILE) $(libdir)/$(SHLIBFILE)
-		ln -sf $(SHLIBFILE) $(libdir)/$(SHLIBSONAME)
-		ln -sf $(SHLIBSONAME) $(libdir)/$(SHLIBFORLINK)
+		$(INSTALL_PROGRAM) $(SHLIBFILE) $(DESTDIR)$(libdir)/$(SHLIBFILE)
+		ln -sf $(SHLIBFILE) $(DESTDIR)$(libdir)/$(SHLIBSONAME)
+		ln -sf $(SHLIBSONAME) $(DESTDIR)$(libdir)/$(SHLIBFORLINK)
 
 uninstall:
 		rm -f $(libdir)/$(SHLIBFILE) $(libdir)/$(SHLIBSONAME)
diff -purwbB adns-1.5.0.orig/src/Makefile.in adns-1.5.0/src/Makefile.in
--- adns-1.5.0.orig/src/Makefile.in	2014-10-20 02:07:03.000000000 +0300
+++ adns-1.5.0/src/Makefile.in	2017-10-23 21:47:49.181299821 +0300
@@ -32,8 +32,10 @@ DIRCFLAGS=	-I. -I$(srcdir)
 install:
 		mkdir -p $(libdir) $(includedir)
 		set -xe; for f in $(TARGETS); \
-			do $(INSTALL_DATA) $$f $(libdir)/$$f; done
-		$(INSTALL_DATA) $(srcdir)/../src/adns.h $(includedir)/adns.h
+			do $(INSTALL_DIR) $(DESTDIR)$(libdir); \
+                           $(INSTALL_DATA) $$f $(DESTDIR)$(libdir)/$$f; done
+		$(INSTALL_DIR)  $(DESTDIR)$(includedir)
+		$(INSTALL_DATA) $(srcdir)/../src/adns.h $(DESTDIR)$(includedir)/adns.h
 
 uninstall:
 		for f in $(TARGETS); do rm -f $(libdir)/$$f; done
