From 99baf1fb0aea45697e447a52ed6754dca6ad787d Mon Sep 17 00:00:00 2001
From: Sami Haapakoski <sami.haapakoski@arm.com>
Date: Thu, 27 May 2021 14:01:24 +0300
Subject: [PATCH] Disable some warnings because they break the build

---
 CMakeLists.txt | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2ab2e01ebf0..6d54df2c124 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -197,9 +197,6 @@ if(CMAKE_COMPILER_IS_GNU)
             set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wformat-signedness")
         endif()
     endif()
-    if (GCC_VERSION VERSION_GREATER 7.0 OR GCC_VERSION VERSION_EQUAL 7.0)
-      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wformat-overflow=2 -Wformat-truncation=2")
-    endif()
     set(CMAKE_C_FLAGS_RELEASE     "-O2")
     set(CMAKE_C_FLAGS_DEBUG       "-O0 -g3")
     set(CMAKE_C_FLAGS_COVERAGE    "-O0 -g3 --coverage")
