diff -Napu thc-hydra-8.3.orig/dpl4hydra.sh thc-hydra-8.3/dpl4hydra.sh
--- thc-hydra-8.3.orig/dpl4hydra.sh	2016-08-11 09:48:55.000000000 +0200
+++ thc-hydra-8.3/dpl4hydra.sh	2017-01-03 10:41:33.273534109 +0100
@@ -8,8 +8,8 @@
 # Credits:  Thanks to van Hauser for support and fixing portability issues.
 #           "The universe is an intelligence test." -Timothy Leary (R.I.P.)
 
-INSTALLDIR=/usr/local
-LOCATION=etc
+INSTALLDIR=
+LOCATION=/etc/hydra
 
 usage ()
 {
Sottodirectory in comune: thc-hydra-8.3.orig/hydra-gtk e thc-hydra-8.3/hydra-gtk
Sottodirectory in comune: thc-hydra-8.3.orig/web e thc-hydra-8.3/web
