--- a/include/tests_kernel	2015-04-16 20:13:59.000000000 +0200
+++ b/include/tests_kernel	2015-05-28 10:05:10.000000000 +0200
@@ -516,7 +516,8 @@
                     FIND=`ls /boot/vmlinuz* 2> /dev/null`
                     if [ ! "${FIND}" = "" ]; then
                          # Display kernels, extract version numbers and sort them numeric per column (up to 6 numbers)
-                         KERNELS=`ls /boot/vmlinuz* | sed 's/vmlinuz-//' | sed 's/\.[a-z].*.//g' | sed 's/-[a-z].*.//g' | sed 's./boot/..' | sed 's/-/./g' | sort -n -k1,1 -k2,2 -k3,3 -k4,4 -k5,5 -k6,6 -t \.`
+                         # Remove generic. and huge. for Slackware machines
+                         KERNELS=`ls /boot/vmlinuz* | sed 's/vmlinuz-//' | sed 's/generic.//' | sed 's/huge.//' | sed 's/\.[a-z].*.//g' | sed 's/-[a-z].*.//g' | sed 's./boot/..' | sed 's/-/./g' | sort -n -k1,1 -k2,2 -k3,3 -k4,4 -k5,5 -k6,6 -t \.`
                       elif [ ! `ls /boot/kernel* 2> /dev/null` = "" ]; then
                          # Display kernels, extract version numbers and sort them numeric per column (up to 6 numbers)
                          # Examples:
