From 6b0ce75347b9b20582e7f1a94d9c266e7c96f48e Mon Sep 17 00:00:00 2001
From: NHellFire <nathan.renniewaldock@gmail.com>
Date: Wed, 11 Jan 2017 07:29:14 +0000
Subject: [PATCH] Fix compilation failure with qt4 after bf8655

---
 src/Logger.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Logger.cpp b/src/Logger.cpp
index 495e28f..e9db8fe 100644
--- a/src/Logger.cpp
+++ b/src/Logger.cpp
@@ -261,7 +261,7 @@ static void qt_logger( QtMsgType msgType,
     {
         fprintf( stderr, "FATAL: %s\n", qPrintable( msg ) );
 
-        if ( msg.contains( "Could not connect to display" ) )
+        if ( ((QString) msg).contains( "Could not connect to display" ) )
             exit( 1 );
         else
             abort();
