From b5a0e2954cd5e27c1e8b47b2e94a84464769fea8 Mon Sep 17 00:00:00 2001
From: Stefan Hundhammer <Stefan.Hundhammer@gmx.de>
Date: Mon, 6 Mar 2017 16:45:16 +0100
Subject: [PATCH] Fixed broken Qt4 build

---
 src/Logger.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/Logger.cpp b/src/Logger.cpp
index 3ee1e65..eff4538 100644
--- a/src/Logger.cpp
+++ b/src/Logger.cpp
@@ -408,9 +408,9 @@ QString Logger::createLogDir( const QString & rawLogDir )
     if ( created )
     {
 	QFile dir( logDir );
-	dir.setPermissions( QFileDevice::ReadOwner  |
-			    QFileDevice::WriteOwner |
-			    QFileDevice::ExeOwner    );
+	dir.setPermissions( QFile::ReadOwner  |
+			    QFile::WriteOwner |
+			    QFile::ExeOwner    );
     }
 
     return logDir;
