diff -Naupr virt-manager-1.4.3/virtinst/osdict.py virt-new/virtinst/osdict.py
--- virt-manager-1.4.3/virtinst/osdict.py	2017-08-16 23:32:14.038542784 +0200
+++ virt-new/virtinst/osdict.py	2017-09-28 16:13:50.563113390 +0200
@@ -159,6 +159,7 @@ class _OSDB(object):
         "rhel5": "rhel5.0",
         "rhel6": "rhel6.0",
         "rhel7": "rhel7.0",
+        "slackware": "slackware14.1"
         "ubuntuhardy": "ubuntu8.04",
         "ubuntuintrepid": "ubuntu8.10",
         "ubuntujaunty": "ubuntu9.04",
@@ -374,7 +375,7 @@ class _OsVariant(object):
         # We should fix this in a new libosinfo version, and then drop
         # this hack
         if self._is_related_to(["fedora24", "rhel7.0", "debian6",
-            "ubuntu13.04", "win8", "win2k12", "mageia5", "centos7.0"],
+            "slackware14.1", "ubuntu13.04", "win8", "win2k12", "mageia5", "centos7.0"],
             check_clones=False, check_derives=False):
             return True
         return False
diff -Naupr virt-manager-1.4.3/virtinst/urlfetcher.py virt-new/virtinst/urlfetcher.py
--- virt-manager-1.4.3/virtinst/urlfetcher.py	2017-09-14 23:49:00.038542784 +0200
+++ virt-new/virtinst/urlfetcher.py	2017-09-28 16:15:23.476951807 +0200
@@ -1346,6 +1346,41 @@ class ALTLinuxDistro(Distro):
         logging.debug("Regex didn't match, not a %s distro", self.name)
         return False
 
+class SlackwareDistro(Distro):
+    # slackware doesn't have installable URLs, so this is just for a
+    # mounted ISO
+    name = "Slackware"
+    urldistro = "slackware"
+    os_variant = "linux"
+
+    _boot_iso_paths = []
+    _xen_kernel_paths = []
+
+    def __init__(self, *args, **kwargs):
+        Distro.__init__(self, *args, **kwargs)
+        if re.match(r'i[4-9]86', self.arch):
+            self.arch = 'i486'
+            self.kname = 'hugesmp.s'
+        else:
+            self.arch = 'x86_64'
+            self.kname = 'huge.s'
+
+        self._hvm_kernel_paths = [("kernels/%s/bzImage" % self.kname,
+                                    "isolinux/initrd.img")]
+
+    def isValidStore(self):
+        # Don't support any paravirt installs
+        if self.type is not None and self.type != "hvm":
+            return False
+
+        # Slackware website / media appear to have a Slackware-HOWTO
+        # file in top level which we can use as our 'magic'
+        # check for validity
+        if not self.fetcher.hasFile("Slackware-HOWTO"):
+            return False
+
+        logging.debug("Regex didn't match, not a %s distro", self.name)
+        return False
 
 # Build list of all *Distro classes
 def _build_distro_list():
