diff -Naupr a/setup.py b/setup.py
--- a/setup.py	2018-04-24 20:11:37.000000000 +0200
+++ b/setup.py	2018-04-25 17:20:52.784697099 +0200
@@ -59,10 +59,9 @@ if len(sys.argv) >= 2 and sys.argv[1] ==
     params = py2exe_params
 else:
     files_spec = [
-        ('etc/bash_completion.d', ['youtube-dl.bash-completion']),
-        ('etc/fish/completions', ['youtube-dl.fish']),
-        ('share/doc/youtube_dl', ['README.txt']),
-        ('share/man/man1', ['youtube-dl.1'])
+        ('/etc/bash_completion.d', ['youtube-dl.bash-completion']),
+        ('/etc/fish/completions', ['youtube-dl.fish']),
+        ('man/man1', ['youtube-dl.1'])
     ]
     root = os.path.dirname(os.path.abspath(__file__))
     data_files = []
