--- suricata.yaml.orig	2015-02-25 14:31:11.000000000 +0100
+++ suricata.yaml.in	2015-02-26 18:47:37.069536431 +0100
@@ -782,12 +782,12 @@ logging:
   # disabled you will get the default - console output.
   outputs:
   - console:
-      enabled: yes
-  - file:
       enabled: no
+  - file:
+      enabled: yes
       filename: /var/log/suricata.log
   - syslog:
-      enabled: no
+      enabled: yes
       facility: local5
       format: "[%i] <%d> -- "
 
@@ -926,47 +926,47 @@ ipfw:
 # if not set, it will look at the current working dir
 default-rule-path: @e_sysconfdir@rules
 rule-files:
- - botcc.rules
- - ciarmy.rules
- - compromised.rules
- - drop.rules
- - dshield.rules
- - emerging-activex.rules
- - emerging-attack_response.rules
- - emerging-chat.rules
- - emerging-current_events.rules
- - emerging-dns.rules
- - emerging-dos.rules
- - emerging-exploit.rules
- - emerging-ftp.rules
- - emerging-games.rules
- - emerging-icmp_info.rules
+# - botcc.rules
+# - ciarmy.rules
+# - compromised.rules
+# - drop.rules
+# - dshield.rules
+# - emerging-activex.rules
+# - emerging-attack_response.rules
+# - emerging-chat.rules
+# - emerging-current_events.rules
+# - emerging-dns.rules
+# - emerging-dos.rules
+# - emerging-exploit.rules
+# - emerging-ftp.rules
+# - emerging-games.rules
+# - emerging-icmp_info.rules
 # - emerging-icmp.rules
- - emerging-imap.rules
- - emerging-inappropriate.rules
- - emerging-malware.rules
- - emerging-misc.rules
- - emerging-mobile_malware.rules
- - emerging-netbios.rules
- - emerging-p2p.rules
- - emerging-policy.rules
- - emerging-pop3.rules
- - emerging-rpc.rules
- - emerging-scada.rules
- - emerging-scan.rules
- - emerging-shellcode.rules
- - emerging-smtp.rules
- - emerging-snmp.rules
- - emerging-sql.rules
- - emerging-telnet.rules
- - emerging-tftp.rules
- - emerging-trojan.rules
- - emerging-user_agents.rules
- - emerging-voip.rules
- - emerging-web_client.rules
- - emerging-web_server.rules
- - emerging-web_specific_apps.rules
- - emerging-worm.rules
+# - emerging-imap.rules
+# - emerging-inappropriate.rules
+# - emerging-malware.rules
+# - emerging-misc.rules
+# - emerging-mobile_malware.rules
+# - emerging-netbios.rules
+# - emerging-p2p.rules
+# - emerging-policy.rules
+# - emerging-pop3.rules
+# - emerging-rpc.rules
+# - emerging-scada.rules
+# - emerging-scan.rules
+# - emerging-shellcode.rules
+# - emerging-smtp.rules
+# - emerging-snmp.rules
+# - emerging-sql.rules
+# - emerging-telnet.rules
+# - emerging-tftp.rules
+# - emerging-trojan.rules
+# - emerging-user_agents.rules
+# - emerging-voip.rules
+# - emerging-web_client.rules
+# - emerging-web_server.rules
+# - emerging-web_specific_apps.rules
+# - emerging-worm.rules
  - tor.rules
  - decoder-events.rules # available in suricata sources under rules dir
  - stream-events.rules  # available in suricata sources under rules dir
