diff -Nur iproute2-2.6.33.orig//Makefile iproute2-2.6.33/Makefile
--- iproute2-2.6.33.orig//Makefile	2010-02-24 21:56:50.000000000 -0600
+++ iproute2-2.6.33/Makefile	2010-02-24 23:00:52.614825494 -0600
@@ -1,14 +1,14 @@
 DESTDIR=/usr/
 ROOTDIR=$(DESTDIR)
-LIBDIR=/usr/lib/
+LIBDIR=@LIBDIR@
 SBINDIR=/sbin
 CONFDIR=/etc/iproute2
-DOCDIR=/share/doc/iproute2
-MANDIR=/share/man
+DOCDIR=/doc/iproute2-@VERSION@
+MANDIR=/man
 ARPDDIR=/var/lib/arpd
 
 # Path to db_185.h include
-DBM_INCLUDE:=$(ROOTDIR)/usr/include
+DBM_INCLUDE:=$(ROOTDIR)/usr/include/db44
 
 SHARED_LIBS = y
 
diff -Nur iproute2-2.6.33.orig//include/iptables.h iproute2-2.6.33/include/iptables.h
--- iproute2-2.6.33.orig//include/iptables.h	2010-02-24 21:56:50.000000000 -0600
+++ iproute2-2.6.33/include/iptables.h	2010-02-24 23:02:21.891841749 -0600
@@ -5,7 +5,7 @@
 #include "libiptc/libiptc.h"
 
 #ifndef IPT_LIB_DIR
-#define IPT_LIB_DIR "/usr/local/lib/iptables"
+#define IPT_LIB_DIR "@LIBDIR@"
 #endif
 
 #ifndef IPPROTO_SCTP
