diff -Naupr a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2019-02-03 12:15:29.000000000 +0100
+++ b/CMakeLists.txt	2019-05-12 13:51:41.975703657 +0200
@@ -100,9 +100,7 @@ cmake_c_add_flags(CMAKE_C_FLAGS -Wall -W
 # We use `override` in certain Qt Classes but the QObject macro does not use it
 # Use compiler option (-Wno-inconsistent-missing-override) to suppress
 # the warning in clang++ 3.6+.
-cmake_c_add_flags(CMAKE_CXX_FLAGS -Wall -Wextra
-  -Wno-deprecated-register -Wno-cast-align
-  -Wno-inconsistent-missing-override)
+cmake_c_add_flags(CMAKE_CXX_FLAGS -Wall -Wextra)
 if(NOT APPLE)
   # -std=c++0x is deprecated but gcc < 4.7 do not recognise c++11 ....
   cmake_c_add_flags(CMAKE_CXX_FLAGS -std=c++0x -D_DEFAULT_SOURCE)
