--- a/Makefile.in.orig	2008-12-30 22:04:25.008554016 +0100
+++ b/Makefile.in	2017-07-13 11:35:43.709803990 +0200
@@ -52,15 +52,15 @@ generate_randfile: generate_randfile.o
 	$(LINK) generate_randfile.o -o generate_randfile
 
 install-bin: $(EXE) $(EXES)
-	mkdir -p $(eprefix)/bin $(eprefix)/sbin
-	@INSTALL_PROGRAM@ @stripping@ $(EXES) $(eprefix)/sbin
-	@INSTALL_PROGRAM@ @stripping@ $(EXE) $(eprefix)/bin
-	@INSTALL_SCRIPT@ $(SCRIPTS) $(eprefix)/bin
+	mkdir -p $(DESTDIR)/$(eprefix)/bin $(DESTDIR)/$(eprefix)/sbin
+	@INSTALL_PROGRAM@ @stripping@ $(EXES) $(DESTDIR)/$(eprefix)/sbin
+	@INSTALL_PROGRAM@ @stripping@ $(EXE) $(DESTDIR)/$(eprefix)/bin
+	@INSTALL_SCRIPT@ $(SCRIPTS) $(DESTDIR)/$(eprefix)/bin
 
 install: install-bin
-	mkdir -p @mandir@/man1 @mandir@/man8
-	@INSTALL_DATA@ $(MAN1) @mandir@/man1
-	@INSTALL_DATA@ $(MAN8) @mandir@/man8
+	mkdir -p $(DESTDIR)@mandir@/man1 $(DESTDIR)@mandir@/man8
+	@INSTALL_DATA@ $(MAN1) $(DESTDIR)@mandir@/man1
+	@INSTALL_DATA@ $(MAN8) $(DESTDIR)@mandir@/man8
 
 %.o: %.cpp
 	$(CXX) -c $<
