From 143044dac548268ba914c8e73ec5ba845a52772a Mon Sep 17 00:00:00 2001
From: Kaiwan N Billimoria <kaiwan.billimoria@gmail.com>
Date: Thu, 13 Feb 2020 18:22:08 +0530
Subject: [PATCH] fortify-file option: fix the 'file does not exist' issue

---
 checksec | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/checksec b/checksec
index 5f3e856..4e47c0b 100755
--- a/checksec
+++ b/checksec
@@ -1752,12 +1752,17 @@ chk_proc_libs () {
 }
 
 chk_fortify_file () {
+  # if first char of pathname is '~' replace it with '${HOME}'
+  if [[ "${CHK_FORTIFY_FILE:0:1}" = '~' ]] ; then
+    CHK_FORTIFY_FILE=${HOME}/${CHK_FORTIFY_FILE:1}
+  fi
+
   if [[ -z "${CHK_FORTIFY_FILE}" ]] ; then
     printf "\033[31mError: Please provide a valid file.\033[m\n\n"
   exit 1
   fi
   # does the file exist?
-  if [[ ! -e "${CHK_FORTIFY_FILE}" ]] ; then
+  if [[ ! -f "${CHK_FORTIFY_FILE}" ]] ; then
     printf "\033[31mError: The file '%s' does not exist.\033[m\n\n" "${CHK_FORTIFY_FILE}"
     exit 1
   fi
