--- configure.orig	2013-07-13 11:44:27.144444982 +0200
+++ configure	2013-08-09 12:08:17.164070336 +0200
@@ -20971,7 +20971,7 @@ if ${ac_cv_lib_memcached_memcached_creat
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lmemcached  $LIBS"
+LIBS="-lmemcached -lpthread $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -21015,7 +21015,7 @@ if test "x$with_libmemcached" = "xyes"
 then
 	BUILD_WITH_LIBMEMCACHED_CPPFLAGS="$with_libmemcached_cppflags"
 	BUILD_WITH_LIBMEMCACHED_LDFLAGS="$with_libmemcached_ldflags"
-	BUILD_WITH_LIBMEMCACHED_LIBS="-lmemcached"
+	BUILD_WITH_LIBMEMCACHED_LIBS="-lmemcached -lpthread"
 
 
 
