diff --git a/src/httperf.c b/src/httperf.c
index e0abd51..63180fa 100755
--- a/src/httperf.c
+++ b/src/httperf.c
@@ -1029,7 +1029,11 @@ main(int argc, char **argv)
                     case 2: 
 #if (OPENSSL_VERSION_NUMBER >= 0x10100000L)
                     ssl_ctx = SSL_CTX_new (TLS_client_method ());
+#if (OPENSSL_VERSION_NUMBER >= 0x10101000L)
                     SSL_CTX_set_options(ssl_ctx, SSL_OP_NO_SSLv3 | SSL_OP_NO_TLSv1 | SSL_OP_NO_TLSv1_1 | SSL_OP_NO_TLSv1_2 | SSL_OP_NO_TLSv1_3);
+#else
+                    SSL_CTX_set_options(ssl_ctx, SSL_OP_NO_SSLv3 | SSL_OP_NO_TLSv1 | SSL_OP_NO_TLSv1_1 | SSL_OP_NO_TLSv1_2);
+#endif
 break;
 #else
                     ssl_ctx = SSL_CTX_new (SSLv2_client_method ()); break;
@@ -1041,7 +1045,11 @@ break;
                     case 3: 
 #if (OPENSSL_VERSION_NUMBER >= 0x10100000L)
                     ssl_ctx = SSL_CTX_new (TLS_client_method ());
+#if (OPENSSL_VERSION_NUMBER >= 0x10101000L)
                     SSL_CTX_set_options(ssl_ctx, SSL_OP_NO_SSLv2 | SSL_OP_NO_TLSv1 | SSL_OP_NO_TLSv1_1 | SSL_OP_NO_TLSv1_2 | SSL_OP_NO_TLSv1_3);
+#else
+                    SSL_CTX_set_options(ssl_ctx, SSL_OP_NO_SSLv2 | SSL_OP_NO_TLSv1 | SSL_OP_NO_TLSv1_1 | SSL_OP_NO_TLSv1_2);
+#endif
 break;
 #else
                     ssl_ctx = SSL_CTX_new (SSLv3_client_method ()); break;

@conraid

Attach files by dragging & dropping,

, or pasting from the clipboard.
Styling with Markdown is supported

