From ea9d65f377c3f7111679e627e10532e5eabea1cb Mon Sep 17 00:00:00 2001
From: Chocobo1 <Chocobo1@users.noreply.github.com>
Date: Sat, 4 Mar 2017 15:03:39 +0800
Subject: [PATCH] Fix incomplete type compile error with Qt4

---
 src/base/http/server.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/base/http/server.cpp b/src/base/http/server.cpp
index 1cc51ee..dd57594 100644
--- a/src/base/http/server.cpp
+++ b/src/base/http/server.cpp
@@ -28,14 +28,18 @@
  * exception statement from your version.
  */
 
+#include "server.h"
+
+#include <QNetworkProxy>
+#include <QStringList>
+
 #ifndef QT_NO_OPENSSL
 #include <QSslSocket>
 #else
 #include <QTcpSocket>
 #endif
-#include <QNetworkProxy>
+
 #include "connection.h"
-#include "server.h"
 
 using namespace Http;
 
