From: Tomasz Buchert <tomasz.buchert@inria.fr>
Date: Sat, 1 Feb 2014 14:26:41 +0100
Subject: TIOCGDEV is not supported by the kernel therefore we comment it out.

---
 src/hd/kbd.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/hd/kbd.c b/src/hd/kbd.c
index ed15656..353f142 100644
--- a/src/hd/kbd.c
+++ b/src/hd/kbd.c
@@ -134,11 +134,13 @@ void add_serial_console(hd_data_t *hd_data)
   }
 
   if(!dev && (fd = open(DEV_CONSOLE, O_RDWR | O_NONBLOCK | O_NOCTTY)) >= 0) {
+    /* Removing since TIOCGDEV isn't supported in the mainline kernel
     if(ioctl(fd, TIOCGDEV, &u) != -1) {
       tty_major = (u >> 8) & 0xfff;
       tty_minor = (u & 0xff) | ((u >> 12) & 0xfff00);
       ADD2LOG(DEV_CONSOLE ": major %u, minor %u\n", tty_major, tty_minor);
     }
+    */
 
     if (0)
 	    ;
