--- Makefile.orig	2013-06-07 15:18:52.000000000 +0200
+++ Makefile	2013-06-20 10:37:07.697044896 +0200
@@ -116,7 +116,7 @@ install: $(TARGET) $(TRANSLATIONS)
 ifneq ($(DEBUG),yes)
 	$(STRIP) $(DESTDIR)/$(BINDIR)/$(TARGET)
 endif
-	cp nl.mo $(DESTDIR)/$(PREFIX)/share/locale/nl/LC_MESSAGES/httping.mo
+	install -D nl.mo $(DESTDIR)/$(PREFIX)/share/locale/nl/LC_MESSAGES/httping.mo
 
 makefile.inc:
 	./configure
