From 4edd5caac653e668511a6d91ea104f3151901085 Mon Sep 17 00:00:00 2001
From: rick <rick@gnous.eu>
Date: Wed, 14 May 2025 20:32:12 +0200
Subject: [PATCH] add missing include

---
 src/exif_entry.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/exif_entry.h b/src/exif_entry.h
index 2ddd240d..5ccbbaaa 100644
--- a/src/exif_entry.h
+++ b/src/exif_entry.h
@@ -3,7 +3,7 @@
 
 #include <vector>
 #include <memory>
-
+#include <cstdint>
 
 
 /**
