diff -Naupr a/config/Makefile.am b/config/Makefile.am
--- a/config/Makefile.am	2024-01-05 22:19:12.000000000 +0100
+++ b/config/Makefile.am	2024-01-09 12:12:59.205801024 +0100
@@ -455,12 +455,12 @@ firewalld.logrotate: firewalld.logrotate
 	$(edit) $< >$@
 
 install-sysconfig:
-	$(MKDIR_P) $(DESTDIR)$(sysconfdir)/sysconfig
-	$(INSTALL_DATA) $(srcdir)/firewalld.sysconfig $(DESTDIR)$(sysconfdir)/sysconfig/firewalld
+	$(MKDIR_P) $(DESTDIR)$(sysconfdir)/default
+	$(INSTALL_DATA) $(srcdir)/firewalld.sysconfig $(DESTDIR)$(sysconfdir)/default/firewalld
 
 uninstall-sysconfig:
-	rm -f $(DESTDIR)$(sysconfdir)/sysconfig/firewalld
-	rmdir $(DESTDIR)$(sysconfdir)/sysconfig || :
+	rm -f $(DESTDIR)$(sysconfdir)/default/firewalld
+	rmdir $(DESTDIR)$(sysconfdir)/default || :
 
 install-rpmmacros:
 	$(MKDIR_P) $(DESTDIR)$(prefix)/lib/rpm/macros.d
diff -Naupr a/config/Makefile.in b/config/Makefile.in
--- a/config/Makefile.in	2024-01-05 22:19:34.000000000 +0100
+++ b/config/Makefile.in	2024-01-09 12:14:01.577737413 +0100
@@ -1158,12 +1158,12 @@ firewalld.logrotate: firewalld.logrotate
 	$(edit) $< >$@
 
 install-sysconfig:
-	$(MKDIR_P) $(DESTDIR)$(sysconfdir)/sysconfig
-	$(INSTALL_DATA) $(srcdir)/firewalld.sysconfig $(DESTDIR)$(sysconfdir)/sysconfig/firewalld
+	$(MKDIR_P) $(DESTDIR)$(sysconfdir)/default
+	$(INSTALL_DATA) $(srcdir)/firewalld.sysconfig $(DESTDIR)$(sysconfdir)/default/firewalld
 
 uninstall-sysconfig:
-	rm -f $(DESTDIR)$(sysconfdir)/sysconfig/firewalld
-	rmdir $(DESTDIR)$(sysconfdir)/sysconfig || :
+	rm -f $(DESTDIR)$(sysconfdir)/default/firewalld
+	rmdir $(DESTDIR)$(sysconfdir)/default || :
 
 install-rpmmacros:
 	$(MKDIR_P) $(DESTDIR)$(prefix)/lib/rpm/macros.d
diff -Naupr a/config/firewalld.init b/config/firewalld.init
--- a/config/firewalld.init	2024-01-05 22:19:12.000000000 +0100
+++ b/config/firewalld.init	2024-01-09 12:14:40.579039995 +0100
@@ -29,7 +29,7 @@ exec="/usr/sbin/firewalld"
 prog="firewalld"
 #config="/etc/firewalld/firewalld.conf"
 
-[ -e /etc/sysconfig/$prog ] && . /etc/sysconfig/$prog
+[ -e /etc/default/$prog ] && . /etc/default/$prog
 
 lockfile=/var/lock/subsys/$prog
 
