diff -Naupr a/config/configure b/config/configure
--- a/config/configure	2020-11-25 21:37:04.000000000 +0100
+++ b/config/configure	2020-11-29 11:36:36.911482110 +0100
@@ -66,9 +66,7 @@ expr "$MAKE_VERSION" '>=' '3.81' || AC_F
 AC_SUB bup_make "$MAKE"
 
 bup_python="$(type -p "$PYTHON")"
-test -z "$bup_python" && bup_python="$(bup_find_prog python2.7 '')"
-test -z "$bup_python" && bup_python="$(bup_find_prog python2.6 '')"
-test -z "$bup_python" && bup_python="$(bup_find_prog python2 '')"
+test -z "$bup_python" && bup_python="$(bup_find_prog python3.9 '')"
 test -z "$bup_python" && bup_python="$(bup_find_prog python3.8 '')"
 test -z "$bup_python" && bup_python="$(bup_find_prog python3.7 '')"
 test -z "$bup_python" && bup_python="$(bup_find_prog python3.6 '')"
