diff -Naupr a/meson.build b/meson.build
--- a/meson.build	1970-01-01 01:00:00.000000000 +0100
+++ b/meson.build	2020-10-26 12:52:39.858986418 +0100
@@ -146,11 +146,11 @@ gjs_dep = dependency('gjs-1.0', version:
 have_gjs = gjs_dep.found()
 
 # Python for optional python dev libs
-pygobject_version = '3.8.0'
+pygobject_version = '3.9.0'
 pygobject_dep = dependency('pygobject-3.0', version: '>= ' + pygobject_version, required: get_option('python'))
 have_python = pygobject_dep.found()
 if have_python
-  python_dep = dependency('python-3.8-embed', version: '>= 3.8')
+  python_dep = dependency('python-3.9-embed', version: '>= 3.9')
 
   version_array = pygobject_version.split('.')
   config_h.set('PYGOBJECT_REQUIRED_MAJOR', version_array[0].to_int())
