diff -Naupr a/configure filezilla-3.67.0/configure
diff -Naupr a/configure.ac filezilla-3.67.0/configure.ac
diff -Naupr a/locales/filezilla.pot filezilla-3.67.0/locales/filezilla.pot
diff -Naupr a/src/engine/controlsocket.cpp filezilla-3.67.0/src/engine/controlsocket.cpp
--- a/src/engine/controlsocket.cpp	2023-06-16 05:30:02.000000000 +0200
+++ filezilla-3.67.0/src/engine/controlsocket.cpp	2024-05-17 05:30:02.000000000 +0200
@@ -1181,7 +1181,7 @@ void CControlSocket::Sleep(fz::duration
 bool CControlSocket::InitBufferPool(bool use_shm)
 {
 	if (!buffer_pool_) {
-		buffer_pool_.emplace(logger_, 8, 0, use_shm);
+		buffer_pool_.emplace(engine_.GetThreadPool(), logger_, 8, 0, use_shm);
 	}
 	return *buffer_pool_;
 }
diff -Naupr a/src/engine/ftp/mkd.cpp filezilla-3.67.0/src/engine/ftp/mkd.cpp
diff -Naupr a/src/interface/cmdline.cpp filezilla-3.67.0/src/interface/cmdline.cpp
diff -Naupr a/src/interface/queue_storage.cpp filezilla-3.67.0/src/interface/queue_storage.cpp
diff -Naupr a/src/interface/verifyhostkeydialog.cpp filezilla-3.67.0/src/interface/verifyhostkeydialog.cpp
diff -Naupr a/src/putty/windows/winsftp.c filezilla-3.67.0/src/putty/windows/winsftp.c
