--- a/src/interface/themeprovider.cpp	2021/07/08 12:15:15	10323
+++ b/src/interface/themeprovider.cpp	2021/07/08 13:00:44	10324
@@ -178,7 +178,7 @@
 
 		// Save in cache
 		if (!cacheFile.empty()) {
-			if (fz::mkdir(fz::to_native(cacheDir.GetPath()), true, true)) {
+			if (fz::mkdir(fz::to_native(cacheDir.GetPath()), true, fz::mkdir_permissions::cur_user)) {
 				image.SaveFile(cacheFile, wxBITMAP_TYPE_PNG);
 			}
 		}
