diff -Naupr a/auparse/test/Makefile.am b/auparse/test/Makefile.am
--- a/auparse/test/Makefile.am	2025-07-10 12:34:13.000000000 +0200
+++ b/auparse/test/Makefile.am	2025-07-25 09:35:56.009283829 +0200
@@ -1,4 +1,4 @@
-# Makefile.am -- 
+# Makefile.am --
 # Copyright 2006-08,2014-17 Red Hat Inc.
 # All Rights Reserved.
 #
@@ -6,25 +6,25 @@
 # modify it under the terms of the GNU Lesser General Public
 # License as published by the Free Software Foundation; either
 # version 2.1 of the License, or (at your option) any later version.
-# 
+#
 # This library is distributed in the hope that it will be useful,
 # but WITHOUT ANY WARRANTY; without even the implied warranty of
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 # Lesser General Public License for more details.
-# 
+#
 # You should have received a copy of the GNU Lesser General Public
 # License along with this library; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
-# 
+#
 # Authors:
 #   Steve Grubb <sgrubb@redhat.com>
 #
 
 CONFIG_CLEAN_FILES = *.loT *.rej *.orig *.cur
-noinst_PROGRAMS = auparse_test auparselol_test lookup_test databuf_test \
+noinst_PROGRAMS = auparse_test auparselol_test lookup_test \
 	auparse_extra_test
 TESTS = run_auparse_tests.sh run_auparselol_test.sh lookup_test \
-	databuf_test auparse_extra_test
+	auparse_extra_test
 dist_check_SCRIPTS = run_auparse_tests.sh run_auparselol_test.sh
 EXTRA_DIST = auparse_test.ref auparse_test.ref.py test.log test2.log test3.log test4.log auditd_raw.sed run_auparse_tests.sh auparse_test.py run_auparselol_test.sh
 CLEANFILES = run_auparse_tests.sh run_auparselol_test.sh
@@ -74,7 +74,7 @@ diffcheck: auparse_test auparselol_test
 	diff -u auparse_test.raw auparse_test.cur
 
 memcheck: auparse_test
-	valgrind --leak-check=yes --show-reachable=yes ./auparse_test 
+	valgrind --leak-check=yes --show-reachable=yes ./auparse_test
 
 pycheck: auparse_test.py
 if USE_PYTHON3
@@ -84,7 +84,7 @@ if USE_PYTHON3
 		srcdir=$(srcdir) $(srcdir)/auparse_test.py
 endif
 
-pydiffcheck: auparse_test.py 
+pydiffcheck: auparse_test.py
 if USE_PYTHON3
 	PYTHONPATH=${top_builddir}/bindings/python/python3/.libs/ \
 	PYTHONDONTWRITEBYTECODE=1 \
