From 3ca45c1a916ac5662142c1cfe7f16fb8d52d4b9f Mon Sep 17 00:00:00 2001
From: Scott Hansen <firecat4153@gmail.com>
Date: Mon, 26 Oct 2015 15:20:41 -0700
Subject: [PATCH] Fixes #28

---
 urlscan/urlscan.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/urlscan/urlscan.py b/urlscan/urlscan.py
index 0233da1..1d9a95c 100644
--- a/urlscan/urlscan.py
+++ b/urlscan/urlscan.py
@@ -244,7 +244,7 @@ def handle_entityref(self, name):
             # added above.
             self.handle_data('&%s;' % name)
 
-urlinternalpattern = r'[{}()@\w/\-%?&.=:;+,#~]'
+urlinternalpattern = r'[{}()@\w/\-%?!&.=:;+,#~]'
 urltrailingpattern = r'[{}()@\w/\-%&=+#]'
 httpurlpattern = (r'(?:(https?|file|ftps?)://' + urlinternalpattern +
                   r'*' + urltrailingpattern + r')')
