diff -Naupr a/src/operations/extract_images.cpp b/src/operations/extract_images.cpp
--- a/src/operations/extract_images.cpp	2022-09-24 10:23:06.000000000 +0200
+++ b/src/operations/extract_images.cpp	2025-03-11 12:43:22.318721834 +0100
@@ -167,12 +167,14 @@ void ExtractImages::extract()
                         // this is to continue extracting images after an error
                         try
                         {
-                            PointerHolder<Buffer> buffer = obj.getStreamData(qpdf_dl_all);
+                            std::shared_ptr<Buffer> buffer = obj.getStreamData(qpdf_dl_all);
+
 
                             int w = obj_dict.getKey("/Width").getUIntValue();
                             int h = obj_dict.getKey("/Height").getUIntValue();
 
-                            Magick::Blob blob{buffer->getBuffer(), buffer->getSize()};
+                            Magick::Blob blob(buffer->getBuffer(), buffer->getSize());
+
                             Magick::Image img;
                             img.read(blob, Magick::Geometry(w, h), depth, map);
                             img.magick(m_image_format.currentText().toStdString());
