diff -Naupr a/src/server/server.c b/src/server/server.c
--- a/src/server/server.c	2020-06-26 01:13:22.000000000 +0200
+++ b/src/server/server.c	2020-08-02 10:09:18.173580642 +0200
@@ -306,14 +306,20 @@ create_response(const char *nurl, const
 					       MHD_RESPMEM_MUST_FREE);
 }
 
-static int cbk_http_request(void *cls,
-			    struct MHD_Connection *connection,
-			    const char *url,
-			    const char *method,
-			    const char *version,
-			    const char *upload_data,
-			    size_t *upload_data_size,
-			    void **ptr)
+
+#if MHD_VERSION < 0x00097002
+static int
+#else
+static enum MHD_Result
+#endif
+cbk_http_request(void *cls,
+		 struct MHD_Connection *connection,
+		 const char *url,
+		 const char *method,
+		 const char *version,
+		 const char *upload_data,
+		 size_t *upload_data_size,
+		 void **ptr)
 {
 	static int dummy;
 	struct MHD_Response *response;
