From a9e2aa0e39cd8eeffd4cccf951b7b91a40f5e8c0 Mon Sep 17 00:00:00 2001
From: esaunders <esaunders@basistech.com>
Date: Fri, 28 Mar 2014 11:13:54 -0400
Subject: [PATCH] Removed unnecessary use of wide strings.

---
 framework/modules/c_LibExifModule/LibExifModule.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/framework/modules/c_LibExifModule/LibExifModule.cpp b/framework/modules/c_LibExifModule/LibExifModule.cpp
index a2b9ffc..41d8ed0 100644
--- a/framework/modules/c_LibExifModule/LibExifModule.cpp
+++ b/framework/modules/c_LibExifModule/LibExifModule.cpp
@@ -297,7 +297,7 @@ extern "C"
     {
         if (pFile == NULL) 
         {
-            LOGERROR(L"ExifExtractModule: passed NULL file pointer.");
+            LOGERROR("ExifExtractModule: passed NULL file pointer.");
             return TskModule::FAIL;
         }
 
@@ -325,7 +325,7 @@ extern "C"
 
             if (exifLoader == NULL)
             {
-                LOGERROR(L"ExifExtractModule - Received NULL ExifLoader pointer");
+                LOGERROR("ExifExtractModule - Received NULL ExifLoader pointer");
                 return TskModule::FAIL;
             }
 
@@ -355,15 +355,15 @@ extern "C"
         }
         catch (TskException& tskEx)
         {
-            std::wstringstream msg;
-            msg << L"ExifExtractModule - Error processing file id " << pFile->getId() << L": " << tskEx.what();
+            std::stringstream msg;
+            msg << "ExifExtractModule - Error processing file id " << pFile->getId() << ": " << tskEx.message();
             LOGERROR(msg.str());
             return TskModule::FAIL;
         }
         catch (std::exception& ex)
         {
-            std::wstringstream msg;
-            msg << L"ExifExtractModule - Error processing file id " << pFile->getId() << L": " << ex.what();
+            std::stringstream msg;
+            msg << "ExifExtractModule - Error processing file id " << pFile->getId() << ": " << ex.what();
             LOGERROR(msg.str());
             return TskModule::FAIL;
         }
