diff -Naupr a/ghostwriter.pro b/ghostwriter.pro
--- a/ghostwriter.pro	2020-02-23 01:49:46.000000000 +0100
+++ b/ghostwriter.pro	2020-11-12 11:08:22.771267002 +0100
@@ -267,7 +267,7 @@ macx {
     appdata.path = $$DATADIR/appdata/
 
     man.files = resources/linux/ghostwriter.1
-    man.path = $$PREFIX/share/man/man1
+    man.path = $$PREFIX/man/man1
 
     qm.files = $${DESTDIR}/translations
     qm.path = $$DATADIR/ghostwriter
