From b09f17a275cede40b52dc411b3ffaa068ec99b44 Mon Sep 17 00:00:00 2001
From: Sebastian Kraft <mail@sebastiankraft.net>
Date: Mon, 11 Aug 2014 21:38:09 +0200
Subject: [PATCH] skip processing if no camera or lens is selected

---
 src/gimplensfun.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/gimplensfun.cpp b/src/gimplensfun.cpp
index 6a66a8f..ea6a5ad 100644
--- a/src/gimplensfun.cpp
+++ b/src/gimplensfun.cpp
@@ -881,6 +881,12 @@ static void process_image (GimpDrawable *drawable) {
     guchar *ImgBuffer;
     guchar *ImgBufferOut;
 
+    if ((sLensfunParameters.CamMaker.length()==0) ||
+        (sLensfunParameters.Camera.length()==0) ||
+        (sLensfunParameters.Lens.length()==0)) {
+            return;
+    }
+
     #ifdef POSIX
     struct timespec profiling_start, profiling_stop;
     #endif
