diff -Naupr a/data/meson.build b/data/meson.build
--- a/data/meson.build	2021-10-07 20:08:49.000000000 +0200
+++ b/data/meson.build	2022-05-07 10:54:44.081273298 +0200
@@ -29,7 +29,6 @@ foreach i : icon_sizes
 endforeach
 
 i18n.merge_file (
-    'desktop',
     input: meson.project_name() + '.desktop.in',
     output: meson.project_name() + '.desktop',
     install: true,
@@ -40,7 +39,6 @@ i18n.merge_file (
 
 
 i18n.merge_file (
-    'appdata',
     input: meson.project_name() + '.appdata.xml.in',
     output: meson.project_name() + '.appdata.xml',
     install: true,
