diff --git a/lib/utils/unsupported.js b/deps/npm/lib/utils/unsupported.js
index 20cee157ee40..aaae8c442665 100644
--- a/lib/utils/unsupported.js
+++ b/lib/utils/unsupported.js
@@ -6,7 +6,8 @@ var supportedNode = [
   {ver: '9', min: '9.0.0'},
   {ver: '10', min: '10.0.0'},
   {ver: '11', min: '11.0.0'},
-  {ver: '12', min: '12.0.0'}
+  {ver: '12', min: '12.0.0'},
+  {ver: '13', min: '13.0.0'}
 ]
 var knownBroken = '<6.0.0'
 
