From bd2c8a47fd04ad38f0b00c7c8648cd723006fd81 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Daniel=20Novomesk=C3=BD?= <dnovomesky@gmail.com>
Date: Tue, 29 Aug 2023 14:03:18 +0200
Subject: [PATCH] Enable building against libavif 1.0.0

---
 CMakeLists.txt       | 5 ++++-
 src/qavifhandler.cpp | 6 +++++-
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f434169..29b67a6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,7 +20,10 @@ include(CheckIncludeFiles)
 set(REQUIRED_QT_VERSION 5.12.0)
 find_package(Qt${QT_MAJOR_VERSION}Gui ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
 
-find_package(libavif 0.8.2 CONFIG)
+find_package(libavif 1.0.0 CONFIG QUIET)
+if(NOT libavif_FOUND)
+  find_package(libavif 0.8.2 CONFIG QUIET)
+endif()
 if (TARGET avif)
   MESSAGE(STATUS "libavif found - we will use dynamic linking")
   set(BUILD_QAVIF_DYNAMIC TRUE)
diff --git a/src/qavifhandler.cpp b/src/qavifhandler.cpp
index e16339d..6e0aba3 100644
--- a/src/qavifhandler.cpp
+++ b/src/qavifhandler.cpp
@@ -359,6 +359,10 @@ bool QAVIFHandler::decode_one_frame()
     avifRGBImage rgb;
     avifRGBImageSetDefaults(&rgb, m_decoder->image);
 
+#if AVIF_VERSION >= 1000000
+    rgb.maxThreads = m_decoder->maxThreads;
+#endif
+
     if (m_decoder->image->depth > 8) {
         rgb.depth = 16;
         rgb.format = AVIF_RGB_FORMAT_RGBA;
@@ -455,7 +459,7 @@ bool QAVIFHandler::decode_one_frame()
     }
 
     if (m_decoder->image->transformFlags & AVIF_TRANSFORM_IMIR) {
-#if AVIF_VERSION > 90100
+#if AVIF_VERSION > 90100 && AVIF_VERSION < 1000000
         switch (m_decoder->image->imir.mode) {
 #else
         switch (m_decoder->image->imir.axis) {
