--- a/CMakeLists.txt  2023-11-08 09:00:12.580968130 -0600
+++ b/CMakeLists.txt  2023-11-08 08:59:46.510691665 -0600
@@ -326,7 +326,7 @@
         message(FATAL_ERROR "** Unable to locate Exiv2... is it installed?")
     else()
         include_directories(${EXIV2_INCLUDE_DIR})
-        target_link_libraries(${PROJECT_NAME} "exiv2lib")
+        target_link_libraries(${PROJECT_NAME} "exiv2")
         add_definitions(-DEXIV2)
         message("** Using Exiv2 ${exiv2_VERSION}")
         if(EXIV2_ENABLE_BMFF)
