diff -Naupr '--exclude=.git' Hand/contrib/libjpeg-turbo/module.defs Hand2/contrib/libjpeg-turbo/module.defs
--- Hand/contrib/libjpeg-turbo/module.defs	2025-04-05 14:11:54.280963973 +0200
+++ Hand2/contrib/libjpeg-turbo/module.defs	2025-04-05 13:57:53.466748047 +0200
@@ -11,7 +11,7 @@ LIBJPEGTURBO.CONFIGURE.args.prefix = -DC
 LIBJPEGTURBO.CONFIGURE.deps        =
 LIBJPEGTURBO.CONFIGURE.static      =
 LIBJPEGTURBO.CONFIGURE.shared      = -DENABLE_SHARED=OFF
-LIBJPEGTURBO.CONFIGURE.extra       = -DCMAKE_INSTALL_LIBDIR=lib
+LIBJPEGTURBO.CONFIGURE.extra       = -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_POLICY_VERSION_MINIMUM=3.5
 
 ifeq (size-aggressive,$(GCC.O))
     LIBJPEGTURBO.CONFIGURE.extra += -DCMAKE_CXX_FLAGS_MINSIZEREL="-Oz -DNDEBUG" -DCMAKE_C_FLAGS_MINSIZEREL="-Oz -DNDEBUG"
diff -Naupr '--exclude=.git' Hand/contrib/libvpl/module.defs Hand2/contrib/libvpl/module.defs
--- Hand/contrib/libvpl/module.defs	2025-04-05 14:11:54.280963973 +0200
+++ Hand2/contrib/libvpl/module.defs	2025-04-05 13:57:53.466748047 +0200
@@ -15,7 +15,7 @@ LIBVPL.CONFIGURE.args.prefix = -DCMAKE_I
 LIBVPL.CONFIGURE.deps        =
 LIBVPL.CONFIGURE.static      =
 LIBVPL.CONFIGURE.shared      = -DBUILD_SHARED_LIBS=OFF
-LIBVPL.CONFIGURE.extra       = -DBUILD_EXPERIMENTAL=ON -DBUILD_TOOLS=OFF -DBUILD_EXAMPLES=OFF
+LIBVPL.CONFIGURE.extra       = -DBUILD_EXPERIMENTAL=ON -DBUILD_TOOLS=OFF -DBUILD_EXAMPLES=OFF -DCMAKE_POLICY_VERSION_MINIMUM=3.5
 
 ifeq ($(GCC.O),$(filter $(GCC.O),size size-aggressive))
     LIBVPL.CONFIGURE.extra += -DCMAKE_BUILD_TYPE=MinSizeRel
diff -Naupr '--exclude=.git' Hand/contrib/svt-av1/module.defs Hand2/contrib/svt-av1/module.defs
--- Hand/contrib/svt-av1/module.defs	2025-04-05 14:11:54.281963955 +0200
+++ Hand2/contrib/svt-av1/module.defs	2025-04-05 13:57:53.466748047 +0200
@@ -16,7 +16,7 @@ SVT-AV1.CONFIGURE.deps        =
 SVT-AV1.CONFIGURE.static      =
 SVT-AV1.CONFIGURE.shared      = -DBUILD_SHARED_LIBS=OFF
 SVT-AV1.CONFIGURE.extra       = -DBUILD_APPS=OFF -DREPRODUCIBLE_BUILDS=ON
-SVT-AV1.CONFIGURE.extra      += -DCMAKE_INSTALL_LIBDIR=lib
+SVT-AV1.CONFIGURE.extra      += -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_POLICY_VERSION_MINIMUM=3.5
 
 ifeq ($(GCC.O),$(filter $(GCC.O),size size-aggressive))
     SVT-AV1.CONFIGURE.extra += -DCMAKE_BUILD_TYPE=MinSizeRel -DMINIMAL_BUILD=ON
diff -Naupr '--exclude=.git' Hand/contrib/x265/A08-cmake.patch Hand2/contrib/x265/A08-cmake.patch
--- Hand/contrib/x265/A08-cmake.patch	1970-01-01 01:00:00.000000000 +0100
+++ Hand2/contrib/x265/A08-cmake.patch	2025-04-05 14:45:24.653803395 +0200
@@ -0,0 +1,19 @@
+diff -Naupr a/source/CMakeLists.txt b/source/CMakeLists.txt
+--- a/source/CMakeLists.txt	2024-11-22 13:07:34.000000000 +0100
++++ b/source/CMakeLists.txt	2025-03-31 13:47:02.835976566 +0200
+@@ -6,15 +6,6 @@ if(NOT CMAKE_BUILD_TYPE)
+         FORCE)
+ endif()
+ message(STATUS "cmake version ${CMAKE_VERSION}")
+-if(POLICY CMP0025)
+-    cmake_policy(SET CMP0025 OLD) # report Apple's Clang as just Clang
+-endif()
+-if(POLICY CMP0042)
+-    cmake_policy(SET CMP0042 NEW) # MACOSX_RPATH
+-endif()
+-if(POLICY CMP0054)
+-    cmake_policy(SET CMP0054 OLD) # Only interpret if() arguments as variables or keywords when unquoted
+-endif()
+ 
+ project (x265)
+ cmake_minimum_required (VERSION 2.8.8) # OBJECT libraries require 2.8.8
diff -Naupr '--exclude=.git' Hand/contrib/x265_10bit/A08-cmake.patch Hand2/contrib/x265_10bit/A08-cmake.patch
--- Hand/contrib/x265_10bit/A08-cmake.patch	1970-01-01 01:00:00.000000000 +0100
+++ Hand2/contrib/x265_10bit/A08-cmake.patch	2025-04-05 13:57:53.466748047 +0200
@@ -0,0 +1,19 @@
+diff -Naupr a/source/CMakeLists.txt b/source/CMakeLists.txt
+--- a/source/CMakeLists.txt	2024-11-22 13:07:34.000000000 +0100
++++ b/source/CMakeLists.txt	2025-03-31 13:47:02.835976566 +0200
+@@ -6,15 +6,6 @@ if(NOT CMAKE_BUILD_TYPE)
+         FORCE)
+ endif()
+ message(STATUS "cmake version ${CMAKE_VERSION}")
+-if(POLICY CMP0025)
+-    cmake_policy(SET CMP0025 OLD) # report Apple's Clang as just Clang
+-endif()
+-if(POLICY CMP0042)
+-    cmake_policy(SET CMP0042 NEW) # MACOSX_RPATH
+-endif()
+-if(POLICY CMP0054)
+-    cmake_policy(SET CMP0054 OLD) # Only interpret if() arguments as variables or keywords when unquoted
+-endif()
+ 
+ project (x265)
+ cmake_minimum_required (VERSION 2.8.8) # OBJECT libraries require 2.8.8
diff -Naupr '--exclude=.git' Hand/contrib/x265_10bit/module.defs Hand2/contrib/x265_10bit/module.defs
--- Hand/contrib/x265_10bit/module.defs	2025-04-05 14:11:54.281963955 +0200
+++ Hand2/contrib/x265_10bit/module.defs	2025-04-05 13:57:53.466748047 +0200
@@ -16,7 +16,7 @@ X265_10.CONFIGURE.args.host   = -DCMAKE_
 X265_10.CONFIGURE.deps        =
 X265_10.CONFIGURE.static      =
 X265_10.CONFIGURE.shared      = -DENABLE_SHARED=OFF
-X265_10.CONFIGURE.extra       = -DENABLE_CLI=OFF -DHIGH_BIT_DEPTH=ON -DEXPORT_C_API=OFF -DCMAKE_ASM_NASM_FLAGS=-w-macro-params-legacy
+X265_10.CONFIGURE.extra       = -DENABLE_CLI=OFF -DHIGH_BIT_DEPTH=ON -DEXPORT_C_API=OFF -DCMAKE_ASM_NASM_FLAGS=-w-macro-params-legacy -DCMAKE_POLICY_VERSION_MINIMUM=3.5
 
 ifneq (1,$(FEATURE.numa))
     X265_10.CONFIGURE.extra  += -DENABLE_LIBNUMA=OFF
diff -Naupr '--exclude=.git' Hand/contrib/x265_12bit/A08-cmake.patch Hand2/contrib/x265_12bit/A08-cmake.patch
--- Hand/contrib/x265_12bit/A08-cmake.patch	1970-01-01 01:00:00.000000000 +0100
+++ Hand2/contrib/x265_12bit/A08-cmake.patch	2025-04-05 13:57:53.466748047 +0200
@@ -0,0 +1,19 @@
+diff -Naupr a/source/CMakeLists.txt b/source/CMakeLists.txt
+--- a/source/CMakeLists.txt	2024-11-22 13:07:34.000000000 +0100
++++ b/source/CMakeLists.txt	2025-03-31 13:47:02.835976566 +0200
+@@ -6,15 +6,6 @@ if(NOT CMAKE_BUILD_TYPE)
+         FORCE)
+ endif()
+ message(STATUS "cmake version ${CMAKE_VERSION}")
+-if(POLICY CMP0025)
+-    cmake_policy(SET CMP0025 OLD) # report Apple's Clang as just Clang
+-endif()
+-if(POLICY CMP0042)
+-    cmake_policy(SET CMP0042 NEW) # MACOSX_RPATH
+-endif()
+-if(POLICY CMP0054)
+-    cmake_policy(SET CMP0054 OLD) # Only interpret if() arguments as variables or keywords when unquoted
+-endif()
+ 
+ project (x265)
+ cmake_minimum_required (VERSION 2.8.8) # OBJECT libraries require 2.8.8
diff -Naupr '--exclude=.git' Hand/contrib/x265_12bit/module.defs Hand2/contrib/x265_12bit/module.defs
--- Hand/contrib/x265_12bit/module.defs	2025-04-05 14:11:54.281963955 +0200
+++ Hand2/contrib/x265_12bit/module.defs	2025-04-05 13:57:53.466748047 +0200
@@ -16,7 +16,7 @@ X265_12.CONFIGURE.args.host   = -DCMAKE_
 X265_12.CONFIGURE.deps        =
 X265_12.CONFIGURE.static      =
 X265_12.CONFIGURE.shared      = -DENABLE_SHARED=OFF
-X265_12.CONFIGURE.extra       = -DENABLE_CLI=OFF -DHIGH_BIT_DEPTH=ON -DEXPORT_C_API=OFF -DMAIN12=ON -DCMAKE_ASM_NASM_FLAGS=-w-macro-params-legacy
+X265_12.CONFIGURE.extra       = -DENABLE_CLI=OFF -DHIGH_BIT_DEPTH=ON -DEXPORT_C_API=OFF -DMAIN12=ON -DCMAKE_ASM_NASM_FLAGS=-w-macro-params-legacy -DCMAKE_POLICY_VERSION_MINIMUM=3.5
 
 ifneq (1,$(FEATURE.numa))
     X265_12.CONFIGURE.extra  += -DENABLE_LIBNUMA=OFF
diff -Naupr '--exclude=.git' Hand/contrib/x265_8bit/A08-cmake.patch Hand2/contrib/x265_8bit/A08-cmake.patch
--- Hand/contrib/x265_8bit/A08-cmake.patch	1970-01-01 01:00:00.000000000 +0100
+++ Hand2/contrib/x265_8bit/A08-cmake.patch	2025-04-05 13:57:53.467748029 +0200
@@ -0,0 +1,19 @@
+diff -Naupr a/source/CMakeLists.txt b/source/CMakeLists.txt
+--- a/source/CMakeLists.txt	2024-11-22 13:07:34.000000000 +0100
++++ b/source/CMakeLists.txt	2025-03-31 13:47:02.835976566 +0200
+@@ -6,15 +6,6 @@ if(NOT CMAKE_BUILD_TYPE)
+         FORCE)
+ endif()
+ message(STATUS "cmake version ${CMAKE_VERSION}")
+-if(POLICY CMP0025)
+-    cmake_policy(SET CMP0025 OLD) # report Apple's Clang as just Clang
+-endif()
+-if(POLICY CMP0042)
+-    cmake_policy(SET CMP0042 NEW) # MACOSX_RPATH
+-endif()
+-if(POLICY CMP0054)
+-    cmake_policy(SET CMP0054 OLD) # Only interpret if() arguments as variables or keywords when unquoted
+-endif()
+ 
+ project (x265)
+ cmake_minimum_required (VERSION 2.8.8) # OBJECT libraries require 2.8.8
diff -Naupr '--exclude=.git' Hand/contrib/x265_8bit/module.defs Hand2/contrib/x265_8bit/module.defs
--- Hand/contrib/x265_8bit/module.defs	2025-04-05 14:11:54.281963955 +0200
+++ Hand2/contrib/x265_8bit/module.defs	2025-04-05 13:57:53.467748029 +0200
@@ -13,7 +13,7 @@ X265_8.CONFIGURE.args.host   = -DCMAKE_H
 X265_8.CONFIGURE.deps        =
 X265_8.CONFIGURE.static      =
 X265_8.CONFIGURE.shared      = -DENABLE_SHARED=OFF
-X265_8.CONFIGURE.extra       = -DENABLE_CLI=OFF -DHIGH_BIT_DEPTH=OFF -DEXTRA_LIB="x265_main10.a;x265_main12.a" -DEXTRA_LINK_FLAGS=-L. -DLINKED_10BIT=ON -DLINKED_12BIT=ON -DCMAKE_ASM_NASM_FLAGS=-w-macro-params-legacy
+X265_8.CONFIGURE.extra       = -DENABLE_CLI=OFF -DHIGH_BIT_DEPTH=OFF -DEXTRA_LIB="x265_main10.a;x265_main12.a" -DEXTRA_LINK_FLAGS=-L. -DLINKED_10BIT=ON -DLINKED_12BIT=ON -DCMAKE_ASM_NASM_FLAGS=-w-macro-params-legacy -DCMAKE_POLICY_VERSION_MINIMUM=3.5 -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_POLICY_DEFAULT_CMP0054=NEW -DCMAKE_POLICY_RELAXED=ON
 
 ifneq (1,$(FEATURE.numa))
     X265_8.CONFIGURE.extra  += -DENABLE_LIBNUMA=OFF
