diff -Naupr a/configure b/configure
--- a/configure	2020-11-24 18:04:03.000000000 +0100
+++ b/configure	2020-11-28 10:51:06.821447996 +0100
@@ -42286,6 +42286,7 @@ rm -f conftest*
     PHP_IMAP_EXPORT void mm_nocritical(void){}
     PHP_IMAP_EXPORT void mm_notify(void){}
     PHP_IMAP_EXPORT void mm_login(void){}
+    PHP_IMAP_EXPORT void mm_login_method(void){}
     PHP_IMAP_EXPORT void mm_diskerror(void){}
     PHP_IMAP_EXPORT void mm_status(void){}
     PHP_IMAP_EXPORT void mm_lsub(void){}
@@ -42346,6 +42347,7 @@ rm -f core conftest.err conftest.$ac_obj
     PHP_IMAP_EXPORT void mm_nocritical(void){}
     PHP_IMAP_EXPORT void mm_notify(void){}
     PHP_IMAP_EXPORT void mm_login(void){}
+    PHP_IMAP_EXPORT void mm_login_method(void){}
     PHP_IMAP_EXPORT void mm_diskerror(void){}
     PHP_IMAP_EXPORT void mm_status(void){}
     PHP_IMAP_EXPORT void mm_lsub(void){}
@@ -42406,6 +42408,7 @@ $as_echo_n "checking whether rfc822_outp
       PHP_IMAP_EXPORT void mm_nocritical(void){}
       PHP_IMAP_EXPORT void mm_notify(void){}
       PHP_IMAP_EXPORT void mm_login(void){}
+      PHP_IMAP_EXPORT void mm_login_method(void){}
       PHP_IMAP_EXPORT void mm_diskerror(void){}
       PHP_IMAP_EXPORT void mm_status(void){}
       PHP_IMAP_EXPORT void mm_lsub(void){}
@@ -42473,6 +42476,7 @@ $as_echo_n "checking whether build with
     PHP_IMAP_EXPORT void mm_nocritical(void){}
     PHP_IMAP_EXPORT void mm_notify(void){}
     PHP_IMAP_EXPORT void mm_login(void){}
+    PHP_IMAP_EXPORT void mm_login_method(void){}
     PHP_IMAP_EXPORT void mm_diskerror(void){}
     PHP_IMAP_EXPORT void mm_status(void){}
     PHP_IMAP_EXPORT void mm_lsub(void){}
diff -Naupr a/ext/imap/config.m4 b/ext/imap/config.m4
--- a/ext/imap/config.m4	2020-11-24 18:04:03.000000000 +0100
+++ b/ext/imap/config.m4	2020-11-28 10:49:15.461875731 +0100
@@ -34,6 +34,7 @@ AC_DEFUN([PHP_IMAP_TEST_BUILD], [
     PHP_IMAP_EXPORT void mm_nocritical(void){}
     PHP_IMAP_EXPORT void mm_notify(void){}
     PHP_IMAP_EXPORT void mm_login(void){}
+    PHP_IMAP_EXPORT void mm_login_method(void){}
     PHP_IMAP_EXPORT void mm_diskerror(void){}
     PHP_IMAP_EXPORT void mm_status(void){}
     PHP_IMAP_EXPORT void mm_lsub(void){}
@@ -262,6 +263,7 @@ if test "$PHP_IMAP" != "no"; then
       PHP_IMAP_EXPORT void mm_nocritical(void){}
       PHP_IMAP_EXPORT void mm_notify(void){}
       PHP_IMAP_EXPORT void mm_login(void){}
+      PHP_IMAP_EXPORT void mm_login_method(void){}
       PHP_IMAP_EXPORT void mm_diskerror(void){}
       PHP_IMAP_EXPORT void mm_status(void){}
       PHP_IMAP_EXPORT void mm_lsub(void){}
diff -Naupr a/ext/imap/php_imap.c b/ext/imap/php_imap.c
--- a/ext/imap/php_imap.c	2020-11-24 18:04:03.000000000 +0100
+++ b/ext/imap/php_imap.c	2020-11-28 10:48:25.836509804 +0100
@@ -4664,7 +4664,7 @@ PHP_IMAP_EXPORT void mm_dlog(char *str)
 	   php debugger? */
 }
 
-PHP_IMAP_EXPORT void mm_login(NETMBX *mb, char *user, char *pwd, long trial)
+PHP_IMAP_EXPORT void mm_login(NETMBX *mb, char *user, char **pwd, long trial)
 {
 
 	if (*mb->user) {
@@ -4672,7 +4672,12 @@ PHP_IMAP_EXPORT void mm_login(NETMBX *mb
 	} else {
 		strlcpy (user, IMAPG(imap_user), MAILTMPLEN);
 	}
-	strlcpy (pwd, IMAPG(imap_password), MAILTMPLEN);
+    *pwd = cpystr(IMAPG(imap_password));    /* alpine-2.22 c-client.a */
+}
+/* alpine-2.22 c-client.a workaround */
+PHP_IMAP_EXPORT void mm_login_method(NETMBX *mb, char *user, void *pwd, long trial, char *method)
+{
+   pwd = NULL;
 }
 
 PHP_IMAP_EXPORT void mm_critical(MAILSTREAM *stream)
