diff -Naupr a/sapi/fpm/php-fpm.conf.in b/sapi/fpm/php-fpm.conf.in
--- a/sapi/fpm/php-fpm.conf.in	2020-11-24 18:04:03.000000000 +0100
+++ b/sapi/fpm/php-fpm.conf.in	2020-11-28 10:43:05.050240412 +0100
@@ -14,14 +14,14 @@
 ; Pid file
 ; Note: the default prefix is @EXPANDED_LOCALSTATEDIR@
 ; Default Value: none
-;pid = run/php-fpm.pid
+pid = run/php-fpm.pid
 
 ; Error log file
 ; If it's set to "syslog", log is sent to syslogd instead of being written
 ; into a local file.
 ; Note: the default prefix is @EXPANDED_LOCALSTATEDIR@
 ; Default Value: log/php-fpm.log
-;error_log = log/php-fpm.log
+error_log = log/php-fpm.log
 
 ; syslog_facility is used to specify what type of program is logging the
 ; message. This lets syslogd specify that messages from different facilities
