From: Rupin Mittal <rupin@apple.com>
Subject: WebAutomationSession::deleteAllCookies must not send null strings over IPC
Origin: https://github.com/WebKit/WebKit/commit/57c2eb4b3e6a46b2f1a03c1fd507ce62c17f7597
Index: webkitgtk/Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp
===================================================================
--- webkitgtk.orig/Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp
+++ webkitgtk/Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp
@@ -1590,6 +1590,7 @@ CommandResult<void> WebAutomationSession
     ASSERT(activeURL.isValid());
 
     String host = activeURL.host().toString();
+    SYNC_FAIL_WITH_PREDEFINED_ERROR_IF(host.isNull(), WindowNotFound);
 
     Ref cookieStore = page->protectedWebsiteDataStore()->cookieStore();
     cookieStore->deleteCookiesForHostnames({ host, domainByAddingDotPrefixIfNeeded(host) }, [] { });
