diff -Naupr a/src/FFmpegWriter.cpp b/src/FFmpegWriter.cpp
--- a/src/FFmpegWriter.cpp	2024-12-21 23:27:30.000000000 +0100
+++ b/src/FFmpegWriter.cpp	2025-08-24 11:00:37.362265249 +0200
@@ -1511,7 +1511,7 @@ void FFmpegWriter::open_video(AVFormatCo
 		switch (video_codec_ctx->codec_id) {
 			case AV_CODEC_ID_H264:
 				video_codec_ctx->max_b_frames = 0;  // At least this GPU doesn't support b-frames
-				video_codec_ctx->profile = FF_PROFILE_H264_BASELINE | FF_PROFILE_H264_CONSTRAINED;
+				video_codec_ctx->profile = AV_PROFILE_H264_BASELINE | AV_PROFILE_H264_CONSTRAINED;
 				av_opt_set(video_codec_ctx->priv_data, "preset", "slow", 0);
 				av_opt_set(video_codec_ctx->priv_data, "tune", "zerolatency", 0);
 				av_opt_set(video_codec_ctx->priv_data, "vprofile", "baseline", AV_OPT_SEARCH_CHILDREN);
