From 14946ed747bc0fe0e7945b4b700d7a9444cb3e46 Mon Sep 17 00:00:00 2001
From: urbanadventurer <horton.nz@gmail.com>
Date: Sat, 9 Jun 2018 05:38:33 +1000
Subject: [PATCH] Fix #244 by adding require 'openssl' for Ruby 2.4 and 2.5
 support.

---
 whatweb | 1 +
 1 file changed, 1 insertion(+)

diff --git a/whatweb b/whatweb
index 2ddf4faf..ee571131 100755
--- a/whatweb
+++ b/whatweb
@@ -47,6 +47,7 @@ require 'rbconfig'  # detect environment, e.g. windows or linux
 require 'resolv'
 require 'resolv-replace' # asynchronous DNS
 require 'open-uri'
+require 'openssl'
 
 
 ## set up load paths - must be before loading lib/ files
