From e7fa0a63f1236b03f643dbb4a77ab0e47e48e188 Mon Sep 17 00:00:00 2001
From: Roman Maksimov <r.l.maksimov@gmail.com>
Date: Mon, 2 Jul 2018 23:53:51 +0300
Subject: [PATCH] Fixed bug in Basic Auth credentials parsing

---
 whatweb | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/whatweb b/whatweb
index ee571131..a8edc497 100755
--- a/whatweb
+++ b/whatweb
@@ -878,8 +878,11 @@ begin
 					raise("Cookie require a parameter, e.g. name=value; name2=value2")
 				end
 			when '-u','--user'
-				$BASIC_AUTH_USER=arg.split(":").first
-				$BASIC_AUTH_PASS=arg.to_s.scan(/^[^:]*:(.+)/).to_s if arg =~ /^[^:]*:(.+)/
+				if arg.include?(':')
+					$BASIC_AUTH_USER, $BASIC_AUTH_PASS = arg.split(":", 2)
+				else
+					raise("Incorrect credentials format.")
+				end
 			when '--debug'
 				$WWDEBUG = true
 			when '--short-help'
