commit ed6eb62f40cb97f71f3df4d982682de68cdf1037
Author: Stefan Jakobs <projects@localside.net>
Date:   Tue May 31 23:34:11 2016 +0200

    support journald as source

diff --git a/scripts/services/syslog-ng b/scripts/services/syslog-ng
old mode 100755
new mode 100644
index dcd1692..d78c835
--- a/scripts/services/syslog-ng
+++ b/scripts/services/syslog-ng
@@ -1,5 +1,5 @@
 ###########################################################################
-# $Id$
+# $Id: syslog-ng 280 2014-12-24 15:29:13Z stefjakobs $
 ###########################################################################
 
 ###########################################################################
@@ -168,7 +168,7 @@ while (defined($ThisLine = <STDIN>)) {
             $Stats_dest{$processed[$i+1]} = 
                $Stats_dest{$processed[$i+1]} + $processed[$i+2];
          } elsif ($processed[$i] eq "source" || $processed[$i] eq "src.internal" ||
-                  $processed[$i] eq 'src.none' ) {
+                  $processed[$i] eq 'src.none' || $processed[$i] eq 'src.journald') {
             $Stats_source{$processed[$i+1]} = 
                $Stats_source{$processed[$i+1]} + $processed[$i+2];
          } elsif ($processed[$i] eq "global") {
@@ -366,7 +366,8 @@ if (keys %Stats_center || keys %Stats_dest || keys %Stats_source ||
       $lost_rcvd = 0 - $Stats_center{received};
       map {
          # skip 'src#X' as this seams to be aggregated into 'src'
-         $lost_rcvd = $lost_rcvd + $Stats_source{$_} unless ($_ =~ /src#\d+/);
+         # skip 'journal' as this is not counted.
+         $lost_rcvd = $lost_rcvd + $Stats_source{$_} unless ($_ =~ /(?:src#\d+|journal)/);
       } keys %Stats_source;
    }
    if ($Stats_center{queued} && %Stats_dest) {
