diff -Naupr a/src/Makefile b/src/Makefile
--- a/src/Makefile	2018-03-19 00:00:42.000000000 +0100
+++ b/src/Makefile	2020-04-13 12:33:02.190608630 +0200
@@ -1,8 +1,9 @@
 CC=gcc
 CFLAGS=-Wall -W -Wmissing-prototypes -D_REENTRANT
 
-INSTLIBDIR=/usr/local/lib
-SYSCONFFILE=/etc/libtrash.conf
+INSTLIBDIR=/usr/lib
+SYSCONFDIR=/etc
+SYSCONFFILE=libtrash.conf
 
 MAJOR =3
 VERSION =3.4
@@ -32,10 +33,11 @@ libtrash.so.${VERSION}: $(SRC) ../libtra
         -o libtrash.so.${VERSION} -ldl
 
 install-libtrash: 
-	TRASH_OFF=YES install libtrash.so.${VERSION} ${INSTLIBDIR}/libtrash.so.${VERSION}
-	TRASH_OFF=YES ln -sf libtrash.so.${VERSION} ${INSTLIBDIR}/libtrash.so
-	TRASH_OFF=YES install libtrash.conf.sys --mode a=r ${SYSCONFFILE}
-	ldconfig	
+	install -d ${DESTDIR}/${INSTLIBDIR}
+	TRASH_OFF=YES install libtrash.so.${VERSION} ${DESTDIR}/${INSTLIBDIR}/libtrash.so.${VERSION}
+	TRASH_OFF=YES ln -sf libtrash.so.${VERSION} ${DESTDIR}/${INSTLIBDIR}/libtrash.so
+	install -d ${DESTDIR}/${SYSCONFDIR}
+	TRASH_OFF=YES install libtrash.conf.sys --mode a=r ${DESTDIR}/${SYSCONFDIR}/${SYSCONFFILE}
 
 clean:
 	rm -f *~
