diff -Naupr a/configure.ac b/configure.ac
--- a/configure.ac	2018-12-27 19:43:22.000000000 +0100
+++ b/configure.ac	2018-12-28 08:12:17.653814049 +0100
@@ -1787,7 +1787,7 @@ AC_MSG_CHECKING(for 'make test' profile)
 if test "$host" != "$build" ; then
     AC_MSG_WARN(Unable to do tests when cross-compiling)
 fi
-
+disable_pcap_findalldevs=no
 dnl Allows user to choose which nic to use for testing purposes
 AC_ARG_WITH(testnic,
     AC_HELP_STRING([--with-testnic=NIC], [Select which network card to use for testing]),
@@ -1840,7 +1840,6 @@ esac])
 
 dnl There's a bug in OS X which causes pcap_findalldevs() to make the wifi NIC to disassociate
 dnl so under OSX we disable the interface list feature
-disable_pcap_findalldevs=no
 osx_frameworks=no
 dnl these need to be dynamic based on OS
 case $host in
