diff --git a/src/app/cli/CMakeLists.txt b/src/app/cli/CMakeLists.txt
index 91c9ff5..79fea6b 100644
--- a/src/app/cli/CMakeLists.txt
+++ b/src/app/cli/CMakeLists.txt
@@ -1,17 +1,41 @@
 ### Check for readline
+set(READLINE_INCLUDE_DIR)
 set(READLINE_LIBRARIES)
 if (WITH_READLINE)
+  find_path(READLINE_INCLUDE_DIR readline/readline.h)
   find_library(READLINE_LIBRARY readline)
-  if (READLINE_LIBRARY)
-    set(READLINE_LIBRARIES ${READLINE_LIBRARY})
+  if (READLINE_INCLUDE_DIR AND READLINE_LIBRARY)
+    # Check whether it needs to be linked against termcap
+    set(_CMAKE_REQUIRED_INCLUDES_TMP "${CMAKE_REQUIRED_INCLUDES}")
+    set(_CMAKE_REQUIRED_LIBRARIES_TMP "${CMAKE_REQUIRED_LIBRARIES}")
+    set(CMAKE_REQUIRED_INCLUDES "${READLINE_INCLUDE_DIR}")
+    set(CMAKE_REQUIRED_LIBRARIES "${READLINE_LIBRARY}")
+    CHECK_CXX_SOURCE_COMPILES("#include<cstdio>\n#include <readline/readline.h>\nint main() {\n  readline(\"\");\n  return 0;\n}\n" READLINE_STANDALONE_FOUND)
+    if (READLINE_STANDALONE_FOUND)
+      set(READLINE_LIBRARIES ${READLINE_LIBRARY})
+    else (READLINE_STANDALONE_FOUND)
+      # Check whether it works with termcap
+      find_library(TERMCAP_LIBRARY termcap)
+      if (TERMCAP_LIBRARY)
+        set(CMAKE_REQUIRED_LIBRARIES "${READLINE_LIBRARY};${TERMCAP_LIBRARY}")
+        CHECK_CXX_SOURCE_COMPILES("#include<cstdio>\n#include <readline/readline.h>\nint main() {\n  readline(\"\");\n  return 0;\n}\n" READLINE_TERMCAP_FOUND)
+        if (READLINE_TERMCAP_FOUND)
+          set(READLINE_LIBRARIES "${READLINE_LIBRARY};${TERMCAP_LIBRARY}")
+        endif (READLINE_TERMCAP_FOUND)
+      endif (TERMCAP_LIBRARY)
+    endif (READLINE_STANDALONE_FOUND)
+    set(CMAKE_REQUIRED_INCLUDES "${_CMAKE_REQUIRED_INCLUDES_TMP}")
+    set(CMAKE_REQUIRED_LIBRARIES "${_CMAKE_REQUIRED_LIBRARIES_TMP}")
+  endif (READLINE_INCLUDE_DIR AND READLINE_LIBRARY)
+  if (READLINE_LIBRARIES)
     message(STATUS "readline found: ${READLINE_LIBRARIES}")
     set(HAVE_READLINE 1)
-  else (READLINE_LIBRARY)
+  else (READLINE_LIBRARIES)
     message(FATAL_ERROR "Could not find readline")
-  endif (READLINE_LIBRARY)
+  endif (READLINE_LIBRARIES)
 endif (WITH_READLINE)
 configure_file(cliconfig.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/cliconfig.h)
-include_directories(${CMAKE_CURRENT_BINARY_DIR})
+include_directories(${CMAKE_CURRENT_BINARY_DIR} ${READLINE_INCLUDE_DIR})
 
 set(cli_SRCS
   maincli.cpp
diff --git a/src/app/cli/readlinecompleter.cpp b/src/app/cli/readlinecompleter.cpp
index a10b0b8..aab1ccc 100644
--- a/src/app/cli/readlinecompleter.cpp
+++ b/src/app/cli/readlinecompleter.cpp
@@ -26,6 +26,7 @@
 
 #include "readlinecompleter.h"
 
+#include <cstdio>
 #include <readline/readline.h>
 #include <cstdlib>
 
diff --git a/src/app/cli/standardinputreader.cpp b/src/app/cli/standardinputreader.cpp
index 691763b..61e31e5 100644
--- a/src/app/cli/standardinputreader.cpp
+++ b/src/app/cli/standardinputreader.cpp
@@ -27,6 +27,7 @@
 #include "standardinputreader.h"
 #include "cliconfig.h"
 #ifdef HAVE_READLINE
+#include <cstdio>
 #include <readline/readline.h>
 #include <readline/history.h>
 #if RL_READLINE_VERSION < 0x0600
