Subject: Fix build failure with GCC 15.1 and glibc 2.41
Bug: https://bitbucket.org/multicoreware/x265_git/issues/1000/
diff --git a/source/dynamicHDR10/json11/json11.cpp b/source/dynamicHDR10/json11/json11.cpp
index 762577735466..74f990a51683 100644
--- a/source/dynamicHDR10/json11/json11.cpp
+++ b/source/dynamicHDR10/json11/json11.cpp
@@ -25,6 +25,7 @@
 #include <cstdlib>
 #include <cstdio>
 #include <limits>
+#include <cstdint>
 
 #if _MSC_VER
 #pragma warning(disable: 4510) //const member cannot be default initialized
