--- ./include/MKhashsize.sh.orig	2019-04-22 14:07:41.451003244 -0500
+++ ./include/MKhashsize.sh	2019-04-22 14:10:09.163003249 -0500
@@ -35,7 +35,7 @@
 echo " */"
 
 test $# = 0 && set Caps
-TABSIZE=`cat "$@" | grep -v '^[ #]' | grep -v "^$" | grep -v "^capalias"| grep -v "^infoalias" | grep -v "^userdef" | grep -v "^used_by" | wc -l`
+TABSIZE=$(cat "$@" | grep -v '^[ #]' | grep -v "^$" | grep -v "^capalias"| grep -v "^infoalias" | grep -v "^userdef" | grep -v "^used_by" | wc -l)
 
 echo ""
 echo "#define CAPTABSIZE	${TABSIZE}"
