--- a/install.sh	2015-02-10 19:46:49.000000000 +0100
+++ b/install.sh	2015-02-23 15:54:23.000000000 +0100
@@ -9,10 +9,10 @@ installShared()
     cp bin/${prg} "${DEST_DIR}${DEST_SHARE}/${prg}"
     chmod 777 "${DEST_DIR}${DEST_SHARE}/${prg}"
     strip     "${DEST_DIR}${DEST_SHARE}/${prg}"
-    chmod 555 "${DEST_DIR}${DEST_SHARE}/${prg}"
+    chmod 755 "${DEST_DIR}${DEST_SHARE}/${prg}"
     echo "#! /bin/sh" > "${DEST_DIR}${DEST_BIN}/${prg}"
     echo "\"${DEST_SHARE}/${prg}\" \"\$@\"" >> "${DEST_DIR}${DEST_BIN}/${prg}"
-    chmod 555 "${DEST_DIR}${DEST_BIN}/${prg}"
+    chmod 755 "${DEST_DIR}${DEST_BIN}/${prg}"
   fi
 }
 
@@ -94,21 +94,21 @@ then
     cp bin/7zCon.sfx "${DEST_DIR}${DEST_SHARE}/7zCon.sfx"
     chmod 777 "${DEST_DIR}${DEST_SHARE}/7zCon.sfx"
     strip     "${DEST_DIR}${DEST_SHARE}/7zCon.sfx"
-    chmod 555 "${DEST_DIR}${DEST_SHARE}/7zCon.sfx"
+    chmod 755 "${DEST_DIR}${DEST_SHARE}/7zCon.sfx"
   fi
 
   if [ -x bin/7z.so ]
   then
     echo "- installing ${DEST_DIR}${DEST_SHARE}/7z.so"
     cp bin/7z.so "${DEST_DIR}${DEST_SHARE}/7z.so"
-    chmod 555 "${DEST_DIR}${DEST_SHARE}/7z.so"
+    chmod 755 "${DEST_DIR}${DEST_SHARE}/7z.so"
   fi
 
   if [ -d bin/Codecs ]
   then
     echo "- installing ${DEST_DIR}${DEST_SHARE}/Codecs"
     cp -r bin/Codecs "${DEST_DIR}${DEST_SHARE}/"
-    chmod 555 "${DEST_DIR}${DEST_SHARE}"/*/*
+    chmod 755 "${DEST_DIR}${DEST_SHARE}"/*/*
   fi
 
 else
@@ -117,7 +117,7 @@ else
     echo "- installing ${DEST_DIR}${DEST_BIN}/7za"
     mkdir -p "${DEST_DIR}${DEST_BIN}"
     cp bin/7za "${DEST_DIR}${DEST_BIN}/7za"
-    chmod 555  "${DEST_DIR}${DEST_BIN}/7za"
+    chmod 755  "${DEST_DIR}${DEST_BIN}/7za"
   fi
 
   if [ -x bin/7zr ]
@@ -125,7 +125,7 @@ else
     echo "- installing ${DEST_DIR}${DEST_BIN}/7zr"
     mkdir -p "${DEST_DIR}${DEST_BIN}"
     cp bin/7zr "${DEST_DIR}${DEST_BIN}/7zr"
-    chmod 555  "${DEST_DIR}${DEST_BIN}/7zr"
+    chmod 755  "${DEST_DIR}${DEST_BIN}/7zr"
   fi
 fi
 
@@ -134,27 +134,27 @@ if [ -d DOC ]
 then
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7z.1"
    sed -e s?"{DEST_SHARE_DOC}"?"${DEST_SHARE_DOC}/DOC"?g man1/7z.1 > "${DEST_DIR}${DEST_MAN}/man1/7z.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7z.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7z.1"
 
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7za.1"
    sed -e s?"{DEST_SHARE_DOC}"?"${DEST_SHARE_DOC}/DOC"?g man1/7za.1 > "${DEST_DIR}${DEST_MAN}/man1/7za.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7za.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7za.1"
 
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7zr.1"
    sed -e s?"{DEST_SHARE_DOC}"?"${DEST_SHARE_DOC}/DOC"?g man1/7zr.1 > "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
 else
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7z.1"
    grep -v "{DEST_SHARE_DOC}" man1/7z.1 > "${DEST_DIR}${DEST_MAN}/man1/7z.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7z.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7z.1"
 
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7za.1"
    grep -v "{DEST_SHARE_DOC}" man1/7za.1 > "${DEST_DIR}${DEST_MAN}/man1/7za.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7za.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7za.1"
 
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7zr.1"
    grep -v "{DEST_SHARE_DOC}" man1/7zr.1 > "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
 fi
 
 if [ -f README ]
@@ -162,7 +162,7 @@ then
   echo "- installing ${DEST_DIR}${DEST_SHARE_DOC}/README"
   mkdir -p  "${DEST_DIR}${DEST_SHARE_DOC}"
   cp README "${DEST_DIR}${DEST_SHARE_DOC}/README"
-  chmod 444 "${DEST_DIR}${DEST_SHARE_DOC}/README"
+  chmod 644 "${DEST_DIR}${DEST_SHARE_DOC}/README"
 fi
 
 if [ -f ChangeLog ]
@@ -170,7 +170,7 @@ then
   echo "- installing ${DEST_DIR}${DEST_SHARE_DOC}/ChangeLog"
   mkdir -p     "${DEST_DIR}${DEST_SHARE_DOC}"
   cp ChangeLog "${DEST_DIR}${DEST_SHARE_DOC}/ChangeLog"
-  chmod 444    "${DEST_DIR}${DEST_SHARE_DOC}/ChangeLog"
+  chmod 644    "${DEST_DIR}${DEST_SHARE_DOC}/ChangeLog"
 fi
 
 if [ -d DOC ]
@@ -178,8 +178,8 @@ then
   echo "- installing HTML help in ${DEST_DIR}${DEST_SHARE_DOC}/DOC"
   mkdir -p "${DEST_DIR}${DEST_SHARE_DOC}"
   cp -r DOC "${DEST_DIR}${DEST_SHARE_DOC}/DOC"
-  find "${DEST_DIR}${DEST_SHARE_DOC}/DOC" -type d -exec chmod 555 {} \;
-  find "${DEST_DIR}${DEST_SHARE_DOC}/DOC" -type f -exec chmod 444 {} \;
+  find "${DEST_DIR}${DEST_SHARE_DOC}/DOC" -type d -exec chmod 755 {} \;
+  find "${DEST_DIR}${DEST_SHARE_DOC}/DOC" -type f -exec chmod 644 {} \;
 fi
 
 use_lang="n"
@@ -188,7 +188,7 @@ if [ -x bin/7zG ]
 then
   use_lang="o"
   cp GUI/p7zipForFilemanager "${DEST_DIR}${DEST_BIN}/"
-  chmod 555 "${DEST_DIR}${DEST_BIN}/"
+  chmod 755 "${DEST_DIR}${DEST_BIN}/"
 fi
 
 if [ -x bin/7zFM ]
@@ -200,7 +200,7 @@ if [ "${use_lang}" = "o" ]
 then
   echo "- installing Lang in ${DEST_DIR}${DEST_SHARE}"
   cp -r GUI/Lang "${DEST_DIR}${DEST_SHARE}/"
-  find "${DEST_DIR}${DEST_SHARE}/Lang" -type d -exec chmod 555 {} \;
-  find "${DEST_DIR}${DEST_SHARE}/Lang" -type f -exec chmod 444 {} \;
+  find "${DEST_DIR}${DEST_SHARE}/Lang" -type d -exec chmod 755 {} \;
+  find "${DEST_DIR}${DEST_SHARE}/Lang" -type f -exec chmod 644 {} \;
 fi
 
