diff -Naupr a/src/backend/jit/llvm/llvmjit_wrap.cpp b/src/backend/jit/llvm/llvmjit_wrap.cpp
--- a/src/backend/jit/llvm/llvmjit_wrap.cpp	2023-09-26 18:21:44.730067546 +0200
+++ b/src/backend/jit/llvm/llvmjit_wrap.cpp	2023-09-26 18:27:32.750203450 +0200
@@ -23,10 +23,11 @@ extern "C"
 
 #include <llvm/IR/Attributes.h>
 #include <llvm/IR/Function.h>
-#include <llvm/MC/SubtargetFeature.h>
 #if LLVM_VERSION_MAJOR > 16
+#include <llvm/TargetParser/SubtargetFeature.h>
 #include <llvm/TargetParser/Host.h>
 #else
+#include <llvm/MC/SubtargetFeature.h>
 #include <llvm/Support/Host.h>
 #endif
 
