diff -Naupr a/src/include/hstr.h b/src/include/hstr.h
--- a/src/include/hstr.h	2023-04-18 23:26:18.000000000 +0200
+++ b/src/include/hstr.h	2024-05-05 10:12:22.363884570 +0200
@@ -26,7 +26,7 @@
 #elif defined(__FreeBSD__)
   #include <ncurses.h>
 #else
-  #include <ncursesw/curses.h>
+  #include <curses.h>
 #endif
 #include <readline/chardefs.h>
 #include <signal.h>
diff -Naupr a/src/include/hstr_curses.h b/src/include/hstr_curses.h
--- a/src/include/hstr_curses.h	2023-04-18 23:26:18.000000000 +0200
+++ b/src/include/hstr_curses.h	2024-05-05 10:12:34.232699920 +0200
@@ -24,7 +24,7 @@
 #elif defined(__FreeBSD__)
 #include <ncurses.h>
 #else
-#include <ncursesw/curses.h>
+#include <curses.h>
 #endif
 
 #define color_attr_on(C) if(terminal_has_colors()) { attron(C); }
