From abe8af5edc797b05569e4f6693883410ca16270f Mon Sep 17 00:00:00 2001
From: Simone Mainardi <mainardi@ntop.org>
Date: Tue, 13 Sep 2016 10:56:02 +0200
Subject: [PATCH] Improves MySQL errors detection

Fixes #733
---
 src/MySQLDB.cpp | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/src/MySQLDB.cpp b/src/MySQLDB.cpp
index 4373c64..a1aab1d 100644
--- a/src/MySQLDB.cpp
+++ b/src/MySQLDB.cpp
@@ -687,13 +687,18 @@ int MySQLDB::exec_sql_query(lua_State *vm, char *sql, bool limitRows) {
     rc = mysql_query(&mysql, sql);
   }
 
-  if((rc != 0) || ((result = mysql_store_result(&mysql)) == NULL)) {
+  if((rc != 0)
+     || (((result = mysql_store_result(&mysql)) == NULL)
+	 && mysql_field_count(&mysql) != 0 /* mysql_store_result() returned nothing; should it have? */)) {
     rc = mysql_errno(&mysql);
 
-    ntop->getTrace()->traceEvent(TRACE_ERROR, "MySQL error: [%s][%d]",
-				 get_last_db_error(&mysql), rc);
+    if(rc) {
+      ntop->getTrace()->traceEvent(TRACE_ERROR, "MySQL error: [%s][%d]",
+				   get_last_db_error(&mysql), rc);
+
+      lua_pushstring(vm, get_last_db_error(&mysql));
+    }
 
-    lua_pushstring(vm, get_last_db_error(&mysql));
     if(m) m->unlock(__FILE__, __LINE__);
     return(rc);
   }
