From 141c778cd94103c876cc75ad2dad74d27ae97c68 Mon Sep 17 00:00:00 2001
From: Simone Mainardi <mainardi@ntop.org>
Date: Fri, 14 Oct 2016 14:47:53 +0200
Subject: [PATCH] Fixes zflow->out_bytes in addFlowStats call

---
 src/NetworkInterface.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/NetworkInterface.cpp b/src/NetworkInterface.cpp
index abc70c7..441f459 100644
--- a/src/NetworkInterface.cpp
+++ b/src/NetworkInterface.cpp
@@ -627,8 +627,8 @@ void NetworkInterface::processFlow(ZMQ_Flow *zflow) {
   flow->addFlowStats(src2dst_direction,
 		     zflow->pkt_sampling_rate*zflow->in_pkts,
 		     zflow->pkt_sampling_rate*zflow->in_bytes, 0,
-		     zflow->pkt_sampling_rate*zflow->out_pkts, 0,
-		     zflow->pkt_sampling_rate*zflow->out_bytes,
+		     zflow->pkt_sampling_rate*zflow->out_pkts,
+		     zflow->pkt_sampling_rate*zflow->out_bytes, 0,
 		     zflow->last_switched);
   p.protocol = zflow->l7_proto, p.master_protocol = NDPI_PROTOCOL_UNKNOWN;
   flow->setDetectedProtocol(p);
