diff -Naupr filezilla-3.10.2-orig/src/interface/Mainfrm.h filezilla-3.10.2/src/interface/Mainfrm.h
--- filezilla-3.10.2-orig/src/interface/Mainfrm.h	2014-12-28 18:48:10.000000000 +0100
+++ filezilla-3.10.2/src/interface/Mainfrm.h	2015-03-04 10:58:10.000000000 +0100
@@ -1,13 +1,18 @@
 #ifndef __MAINFRM_H__
 #define __MAINFRM_H__
 
+#include "statusbar.h"
+#include "engine_context.h"
+
+#include <wx/timer.h>
+
 #ifndef __WXMAC__
 #include <wx/taskbar.h>
 #endif
 
-#include "statusbar.h"
-
-#include "engine_context.h"
+#if FZ_MANUALUPDATECHECK
+#include "updater.h"
+#endif
 
 class CAsyncRequestQueue;
 class CContextControl;
@@ -25,13 +30,7 @@ class CThemeProvider;
 class CToolBar;
 class CWindowStateManager;
 
-
-#if FZ_MANUALUPDATECHECK
-#include "updater.h"
-#endif
-class CWindowStateManager;
-
-class CMainFrame : public wxNavigationEnabled<wxFrame>
+class CMainFrame final : public wxNavigationEnabled<wxFrame>
 #if FZ_MANUALUPDATECHECK
 	, protected CUpdateHandler
 #endif
