diff -Naupr a/src/common/colorspaces.h b/src/common/colorspaces.h
--- a/src/common/colorspaces.h	2025-02-08 18:55:20.000000000 +0100
+++ b/src/common/colorspaces.h	2025-06-01 13:36:21.566083127 +0200
@@ -1,6 +1,6 @@
 /*
     This file is part of darktable,
-    Copyright (C) 2010-2024 darktable developers.
+    Copyright (C) 2010-2025 darktable developers.
 
     darktable is free software: you can redistribute it and/or modify
     it under the terms of the GNU General Public License as published by
@@ -22,6 +22,8 @@
 
 #include <lcms2.h>
 
+G_BEGIN_DECLS
+
 // this was removed from lcms2 in 2.4
 #ifndef TYPE_XYZA_FLT
   #define TYPE_XYZA_FLT (FLOAT_SH(1)|COLORSPACE_SH(PT_XYZ)|EXTRA_SH(1)|CHANNELS_SH(3)|BYTES_SH(4))
@@ -419,8 +421,11 @@ void dt_make_transposed_matrices_from_pr
                                                                const float whitepoint[2],
                                                                dt_colormatrix_t RGB_to_XYZ_transposed);
 
+G_END_DECLS
+
 // clang-format off
 // modelines: These editor modelines have been set for all relevant files by tools/update_modelines.py
 // vim: shiftwidth=2 expandtab tabstop=2 cindent
 // kate: tab-indents: off; indent-width 2; replace-tabs on; indent-mode cstyle; remove-trailing-spaces modified;
 // clang-format on
+
diff -Naupr a/src/imageio/format/imageio_format_api.h b/src/imageio/format/imageio_format_api.h
--- a/src/imageio/format/imageio_format_api.h	2025-02-08 18:55:20.000000000 +0100
+++ b/src/imageio/format/imageio_format_api.h	2025-06-01 13:35:15.618272875 +0200
@@ -1,6 +1,6 @@
 /*
     This file is part of darktable,
-    Copyright (C) 2016-2021 darktable developers.
+    Copyright (C) 2016-2025 darktable developers.
 
     darktable is free software: you can redistribute it and/or modify
     it under the terms of the GNU Lesser General Public License as published by
@@ -20,10 +20,6 @@
 
 #ifdef FULL_API_H
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include <stddef.h>
 #include <stdint.h>
 
@@ -33,6 +29,8 @@ struct dt_dev_pixelpipe_t;
 
 #include "common/colorspaces.h" // because forward declaring enums doesn't work in C++ :(
 
+G_BEGIN_DECLS
+
 /* early definition of modules to do type checking */
 
 #pragma GCC visibility push(default)
@@ -96,9 +94,7 @@ OPTIONAL(int, read_image, struct dt_imag
 
 #pragma GCC visibility pop
 
-#ifdef __cplusplus
-}
-#endif
+G_END_DECLS
 
 #endif // FULL_API_H
 
@@ -107,3 +103,4 @@ OPTIONAL(int, read_image, struct dt_imag
 // vim: shiftwidth=2 expandtab tabstop=2 cindent
 // kate: tab-indents: off; indent-width 2; replace-tabs on; indent-mode cstyle; remove-trailing-spaces modified;
 // clang-format on
+
diff -Naupr a/src/lua/lua.h b/src/lua/lua.h
--- a/src/lua/lua.h	2025-02-08 18:55:20.000000000 +0100
+++ b/src/lua/lua.h	2025-06-01 13:37:04.530309550 +0200
@@ -1,6 +1,6 @@
 /*
    This file is part of darktable,
-   Copyright (C) 2013-2024 darktable developers.
+   Copyright (C) 2013-2025 darktable developers.
 
    darktable is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -27,7 +27,12 @@
 #include "common/dtpthread.h"
 #include <glib.h>
 
+// while it's poor practice to have includes inside an "extern C" block, we need to
+// do this here because the Lua headers don't supply it when compiling under C++....
+G_BEGIN_DECLS
+
 #ifdef USE_LUA
+
 #include <lautoc.h>
 #include <lauxlib.h>
 #include <lua.h>
@@ -97,9 +102,12 @@ typedef struct
 } dt_lua_state_t;
 #endif
 
+G_END_DECLS
+
 // clang-format off
 // modelines: These editor modelines have been set for all relevant files by tools/update_modelines.py
 // vim: shiftwidth=2 expandtab tabstop=2 cindent
 // kate: tab-indents: off; indent-width 2; replace-tabs on; indent-mode cstyle; remove-trailing-spaces modified;
 // clang-format on
 
+
