diff -Naupr a/tools/make-version.py b/tools/make-version.py
--- a/tools/make-version.py	2022-07-28 21:05:42.000000000 +0200
+++ b/tools/make-version.py	2022-07-29 15:21:48.268600658 +0200
@@ -270,7 +270,7 @@ def read_git_archive(tagged_version_extr
         match = re.match(r'^tag: (v[1-9].+)', git_ref)
         if match:
             is_tagged = True
-            vcs_tag = match.groups[0]
+            vcs_tag = match.groups(0)
 
     if is_tagged:
         print(f"We are on tag {vcs_tag}.")
