diff -Naupr wxWidgets-3.1.0-orig/src/common/appbase.cpp wxWidgets-3.1.0/src/common/appbase.cpp
--- wxWidgets-3.1.0-orig/src/common/appbase.cpp	2016-04-16 08:49:51.230541137 +0200
+++ wxWidgets-3.1.0/src/common/appbase.cpp	2016-04-16 08:51:01.000000000 +0200
@@ -846,10 +846,7 @@ bool wxAppConsoleBase::CheckBuildOptions
         msg.Printf(wxT("Mismatch between the program and library build versions detected.\nThe library used %s,\nand %s used %s."),
                    lib.c_str(), progName.c_str(), prog.c_str());
 
-        wxLogFatalError(msg.c_str());
-
-        // normally wxLogFatalError doesn't return
-        return false;
+        wxLogWarning(msg.c_str());
     }
 
     return true;
