From 1826f8549c4d606695f4426b2c267dddcdb95c7e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alexander=20K=C3=B6ppe?= <format_c@online.de>
Date: Mon, 25 Nov 2019 18:53:55 +0100
Subject: [PATCH] Fix segmentation fault when parsing NTLM/HTTP handshake
 (#960)

---
 src/dissectors/ec_http.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dissectors/ec_http.c b/src/dissectors/ec_http.c
index 9e47e1a92..6384d0afd 100644
--- a/src/dissectors/ec_http.c
+++ b/src/dissectors/ec_http.c
@@ -501,7 +501,7 @@ static int Parse_NTLM_Auth(char *ptr, char *from_here, struct packet_object *po)
           */  
          if (conn_status->c_status == NTLM_WAIT_RESPONSE) {
             /* Fill the user and passwords */
-            response_struct  = (tSmbNtlmAuthResponse *) to_decode;
+            response_struct  = (tSmbNtlmAuthResponse *) decoded;
             po->DISSECTOR.user = strdup(GetUnicodeString(response_struct, uUser));
             SAFE_CALLOC(po->DISSECTOR.pass, strlen(po->DISSECTOR.user) + 150, sizeof(char));
             snprintf(po->DISSECTOR.pass, strlen(po->DISSECTOR.user) + 150, "(NTLM) %s:\"\":\"\":", po->DISSECTOR.user);
