From 232b894f505d0274d11b356370b6b1b0bb875d24 Mon Sep 17 00:00:00 2001
From: Alexander Koeppe <format_c@online.de>
Date: Sun, 24 Nov 2019 20:04:50 +0100
Subject: [PATCH] PCAP_TIMEOUT 0 for Linux caused issues with libpcap>=1.9.1

---
 src/ec_network.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ec_network.c b/src/ec_network.c
index a189d2452..dc748a5eb 100644
--- a/src/ec_network.c
+++ b/src/ec_network.c
@@ -11,14 +11,14 @@
 #include <ifaddrs.h>
 #endif
 
-#if defined(OS_BSD_OPEN) || defined(OS_LINUX)
-   /* LINUX does not care about timeout */
+#if defined(OS_BSD_OPEN)
    /* OPENBSD needs 0 */
    #define PCAP_TIMEOUT 0
 #elif defined(OS_SOLARIS)
    /* SOLARIS needs > 1 */
    #define PCAP_TIMEOUT 10
 #else
+   /* LINUX needs > 0 */
    /* FREEBSD needs 1 */
    /* MACOSX  needs 1 */
    #define PCAP_TIMEOUT 1
