From 2474c9eb4522ad439b817785321e3fd0c105c4e7 Mon Sep 17 00:00:00 2001
From: Alexander Koeppe <format_c@online.de>
Date: Thu, 21 Nov 2019 18:20:56 +0100
Subject: [PATCH] update Changelog with SSL improvements

---
 CHANGELOG | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CHANGELOG b/CHANGELOG
index 277250543..4b6b5d597 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -3,6 +3,11 @@ Legend:
          + new feature
          - old feature removed
          !! bug fixed
+=========================================
+x.X.X-XXXXXXXXX    YYYYMMDD
+   !! Fix TLS 1.3 interception issues (replace fake certificate with proper key length)
+    + Take over client-side SNI extension in ClientHello in SSL interception (req. OpenSSL 1.1.1)
+    + Take over SAN certificate extension from server certificate in SSL interception
 
 =========================================
 0.8.4-XXXXXXXXX    00000000
