From 2ef313ac68818a6c87e72d0b93e643518ee9de3a Mon Sep 17 00:00:00 2001
From: Alexander Koeppe <alexander@koeppe.rocks>
Date: Tue, 19 Nov 2019 10:16:14 +0100
Subject: [PATCH] Remove Easter Egg

---
 src/ec_main.c | 30 ------------------------------
 1 file changed, 30 deletions(-)

diff --git a/src/ec_main.c b/src/ec_main.c
index cbad1d19d..f6ef65b11 100644
--- a/src/ec_main.c
+++ b/src/ec_main.c
@@ -51,8 +51,6 @@
 
 /* protos */
 
-static void time_check(void);
-
 /*******************************************/
 
 int main(int argc, char *argv[])
@@ -86,9 +84,6 @@ int main(int argc, char *argv[])
    /* getopt related parsing...  */
    parse_options(argc, argv);
 
-   /* check the date */
-   time_check();
-
    /* 
     * get the list of available interfaces 
     * 
@@ -217,31 +212,6 @@ int main(int argc, char *argv[])
    return 0; //Never reaches here
 }
 
-static void time_check(void)
-{
-   /* 
-    * a nice easter egg... 
-    * just to waste some time of code reviewers... ;) 
-    * ALoR, keeping this for you buddy! :)
-    *
-    * trust me, it's not evil ;) only a boring afternoon, and nothing to do...
-    */
-   time_t K9=time(NULL);char G5P[1<<6],*o=G5P,*O;uint U4M, _,__=0; char dMG[]= 
-   "\n*\n^1U4Mm\x04wW#K\x2e\x0e+X\x7f\f,N'U!I-L5?";struct{char X5T[7];int dMG;
-   int U4M;} X5T[]={{"N!WwFr", 0x414c6f52,0},{"S6FfUe", 0x4e614741,0}};sprintf
-   (G5P,"%s",ctime(&K9));o+=4;O=strchr(o+4,' ');*O=0; for(U4M=(1<<5)-(1<<2)+1;
-   U4M>0;U4M--){dMG[U4M]=dMG[U4M]^dMG[U4M-1];}for(U4M=0;U4M<sizeof(X5T)/sizeof
-   (*X5T);U4M++){for(_=(1<<2)+1; _>0;_--){X5T[U4M].X5T[_]=X5T[U4M].X5T[_]^X5T[
-   U4M].X5T[_-1];}if(!strcmp(X5T[U4M].X5T,o)){char T0Q[]="\n\0O!M4\x14r\x1doO"
-   ";T0Q(\bm\x19m\bz\x19x\b(A2\x12s\x1d=X5T=Q&G5Pp\x03l\n~\th\x1a\x7f_dMG\x06"
-   "hH-@" "!H$\x04s\x1av\x1a:X=\x1d|\f|\x0ek\ba\0t\x11u[u[{^-m\fb\x16\x7f\x19"
-   "v\x04oA\x2e\\;1;K9\\/\\|9w#f4\x1a\x34\x1a\x1a";for(_=(1<<7)-(1<<3)-(1<<2)+
-   1;_>0;_--){T0Q[_]=T0Q[_]^T0Q[_-1];}write(1,dMG,1);while(__++<1<<5)printf(""
-   "%c",(1<<5)+(1<<3)+(1<<1));X5T[U4M].dMG=ntohl(X5T[U4M].dMG);printf(dMG,&X5T
-   [U4M].dMG);while(--__){printf("%c",(1<<6)-(1<<4)-(1<<3)+(1<<1));}printf(T0Q
-   ,&X5T[U4M].dMG);getchar();break;}}
-}
-
 /* EOF */
 
 // vim:ts=3:expandtab
