diff -Naupr a/cmake/Modules/FindGTK3.cmake b/cmake/Modules/FindGTK3.cmake
--- a/cmake/Modules/FindGTK3.cmake	2020-08-01 14:32:06.000000000 -0000
+++ b/cmake/Modules/FindGTK3.cmake	2020-08-02 07:32:00.737760780 -0000
@@ -409,7 +409,7 @@ endif()
 #
 
 find_package(Freetype QUIET)
-list(APPEND GTK3_INCLUDE_DIRS ${FREETYPE_INCLUDE_DIRS})
+list(APPEND GTK3_INCLUDE_DIRS ${FREETYPE_INCLUDE_DIRS} /usr/include/harfbuzz)
 list(APPEND GTK3_LIBRARIES ${FREETYPE_LIBRARIES})
 
 foreach(_GTK3_component ${GTK3_FIND_COMPONENTS})
