From db7965f98e8e0bc70810b01cb833e64c7f03baba Mon Sep 17 00:00:00 2001
From: Alexander Koeppe <format_c@online.de>
Date: Thu, 21 Nov 2019 18:06:01 +0100
Subject: [PATCH] explicitely enable ECDH key exchange capabilities (requires
 OpenSSL 1.0.2)

---
 src/ec_sslwrap.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/ec_sslwrap.c b/src/ec_sslwrap.c
index 01aa0c6c4..bcd8eb8bd 100644
--- a/src/ec_sslwrap.c
+++ b/src/ec_sslwrap.c
@@ -58,6 +58,10 @@
 #define HAVE_OPAQUE_RSA_DSA_DH 1 /* since 1.1.0 -pre5 */
 #endif
 
+#if (OPENSSL_VERSION_NUMBER >= 0x10002000L)
+#define HAVE_OPENSSL_1_0_2
+#endif
+
 #if (OPENSSL_VERSION_NUMBER >= 0x10101000L)
 #define HAVE_OPENSSL_1_1_1
 #endif
@@ -1095,6 +1099,12 @@ static void sslw_init(void)
    ssl_ctx_client = SSL_CTX_new(TLS_server_method());
    ssl_ctx_server = SSL_CTX_new(TLS_client_method());
 
+#ifdef HAVE_OPENSSL_1_0_2
+   SSL_CTX_set_ecdh_auto(ssl_ctx_client, 1);
+   SSL_CTX_set_ecdh_auto(ssl_ctx_server, 1);
+#endif
+
+
    ON_ERROR(ssl_ctx_client, NULL, "Could not create client SSL CTX");
    ON_ERROR(ssl_ctx_server, NULL, "Could not create server SSL CTX");
 
