diff -Naupr a/app/dialogs/about-dialog.c b/app/dialogs/about-dialog.c
--- a/app/dialogs/about-dialog.c	2023-07-05 16:00:47.000000000 +0200
+++ b/app/dialogs/about-dialog.c	2023-07-10 11:58:44.123362142 +0200
@@ -178,7 +178,9 @@ about_dialog_create (GimpCoreConfig *con
 #ifdef GIMP_UNSTABLE
           about_dialog_add_unstable_message (children->data);
 #endif /* GIMP_UNSTABLE */
+#ifdef CHECK_UPDATE
           about_dialog_add_update (&dialog, config);
+#endif /* CHECK_UPDATE */
         }
       else
         g_warning ("%s: ooops, no box in this container?", G_STRLOC);
