--- a/src/zmap.c	2015-02-22 10:16:15.121931683 +0100
+++ b/src/zmap.c	2015-02-22 10:13:56.063670150 +0100
@@ -20,6 +20,10 @@
 
 #include <pcap/pcap.h>
 
+#ifdef JSON
+#include <json.h>
+#endif
+
 #include <pthread.h>
 
 #include "../lib/includes.h"
@@ -602,7 +606,6 @@ int main(int argc, char *argv[])
 
     if (args.user_metadata_given) {
 #ifdef JSON
-#include <json.h>
         zconf.custom_metadata_str = args.user_metadata_arg;
         if (!json_tokener_parse(zconf.custom_metadata_str)) {
             log_fatal("metadata", "unable to parse custom user metadata");
 #ifndef PFRING
 	// Set the correct number of threads, default to num_cores - 1
 	if (args.sender_threads_given) {

