diff -Naupr valgrind-3.10.1-orig/configure valgrind-3.10.1/configure
--- valgrind-3.10.1-orig/configure	2014-11-25 20:42:25.000000000 +0100
+++ valgrind-3.10.1/configure	2015-05-13 17:42:41.000000000 +0200
@@ -5553,8 +5553,8 @@ $as_echo_n "checking for the kernel vers
         kernel=`uname -r`
 
         case "${kernel}" in
-             2.6.*|3.*)
-        	    { $as_echo "$as_me:${as_lineno-$LINENO}: result: 2.6.x/3.x family (${kernel})" >&5
+             2.6.*|3.*|4.*)
+        	    { $as_echo "$as_me:${as_lineno-$LINENO}: result: 2.6.x/3.x/4.x family (${kernel})" >&5
 $as_echo "2.6.x/3.x family (${kernel})" >&6; }
 
 $as_echo "#define KERNEL_2_6 1" >>confdefs.h
@@ -5572,7 +5572,7 @@ $as_echo "#define KERNEL_2_4 1" >>confde
              *)
         	    { $as_echo "$as_me:${as_lineno-$LINENO}: result: unsupported (${kernel})" >&5
 $as_echo "unsupported (${kernel})" >&6; }
-        	    as_fn_error $? "Valgrind works on kernels 2.4, 2.6" "$LINENO" 5
+        	    as_fn_error $? "Valgrind works on kernels 2.4, 2.6, 3.x, 4x" "$LINENO" 5
         	    ;;
         esac
 
@@ -6840,6 +6840,26 @@ $as_echo "#define GLIBC_2_20 1" >>confde
 
 	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
+     2.20)
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: 2.20 family" >&5
+$as_echo "2.20 family" >&6; }
+
+$as_echo "#define GLIBC_2_20 1" >>confdefs.h
+
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
+     2.21)
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: 2.21 family" >&5
+$as_echo "2.21 family" >&6; }
+
+$as_echo "#define GLIBC_2_21 1" >>confdefs.h
+
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
 	;;
      darwin)
diff -Naupr valgrind-3.10.1-orig/configure.ac valgrind-3.10.1/configure.ac
--- valgrind-3.10.1-orig/configure.ac	2014-11-25 20:41:13.000000000 +0100
+++ valgrind-3.10.1/configure.ac	2015-05-13 17:43:14.000000000 +0200
@@ -303,8 +303,8 @@ case "${host_os}" in
         kernel=`uname -r`
 
         case "${kernel}" in
-             2.6.*|3.*) 
-        	    AC_MSG_RESULT([2.6.x/3.x family (${kernel})])
+             2.6.*|3.*|4.*) 
+        	    AC_MSG_RESULT([2.6.x/3.x/4.x family (${kernel})])
         	    AC_DEFINE([KERNEL_2_6], 1, [Define to 1 if you're using Linux 2.6.x or Linux 3.x])
         	    ;;
 
@@ -315,7 +315,7 @@ case "${host_os}" in
 
              *) 
         	    AC_MSG_RESULT([unsupported (${kernel})])
-        	    AC_MSG_ERROR([Valgrind works on kernels 2.4, 2.6])
+        	    AC_MSG_ERROR([Valgrind works on kernels 2.4, 2.6, 3.x, 4.x])
         	    ;;
         esac
 
@@ -1017,6 +1017,20 @@ case "${GLIBC_VERSION}" in
 	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
+     2.20)
+	AC_MSG_RESULT(2.20 family)
+	AC_DEFINE([GLIBC_2_20], 1, [Define to 1 if you're using glibc 2.20.x])
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
+     2.21)
+	AC_MSG_RESULT(2.21 family)
+	AC_DEFINE([GLIBC_2_21], 1, [Define to 1 if you're using glibc 2.21.x])
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
 	;;
      darwin)
 	AC_MSG_RESULT(Darwin)
