From 7cc2a08ef4319fcf4756da923d560c027fe9b77a Mon Sep 17 00:00:00 2001
From: Werner Fink <werner@suse.de>
Date: Wed, 1 Jun 2016 14:19:09 +0100
Subject: [PATCH] fix crash when the main mailbox doesn't exist

fixes: #39
---
 headers.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/headers.c b/headers.c
index 465bcf3..1fa7d8e 100644
--- a/headers.c
+++ b/headers.c
@@ -357,6 +357,9 @@ void mutt_scan_labels(CONTEXT *ctx)
 {
   int i;
 
+  if (!ctx)
+    return;
+
   for (i = 0; i < ctx->msgcount; i++)
     if (ctx->hdrs[i]->env->labels)
       mutt_label_ref_inc(ctx->hdrs[i]->env);
