From 2517004bb3c48653100f0c6a6da16fde7927755e Mon Sep 17 00:00:00 2001
From: Jonathan Hudson <jh+github@daria.co.uk>
Date: Sat, 12 Apr 2025 19:28:51 +0100
Subject: [PATCH] remove compact from settings class

---
 src/settings.vala | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/settings.vala b/src/settings.vala
index 220b91a..d2660f9 100644
--- a/src/settings.vala
+++ b/src/settings.vala
@@ -1,4 +1,3 @@
-[Compact]
 public class Conf : Object {
     public string audio_device { get; set; }
     public string video_dir { get; set; }
