Index: /trunk/pokerth/src/net/common/ircthread.cpp
===================================================================
--- /trunk/pokerth/src/net/common/ircthread.cpp (revision 2899)
+++ /trunk/pokerth/src/net/common/ircthread.cpp (revision 3173)
@@ -21,4 +21,5 @@
 #include <net/socket_msg.h>
 #include <libircclient/libircclient.h>
+#include <libircclient/libirc_rfcnumeric.h>
 #include <boost/algorithm/string/predicate.hpp>
 #include <queue>
