diff -Naupr a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2022-09-27 02:28:46.000000000 +0200
+++ b/CMakeLists.txt	2025-09-07 18:34:40.155646393 +0200
@@ -1,7 +1,7 @@
-cmake_minimum_required(VERSION 3.2)
+cmake_minimum_required(VERSION 3.5)
 project(pystring CXX)
 
-set(BUILD_SHARED_LIBS YES)
+option (BUILD_SHARED_LIBS "Build shared libraries (set to OFF to build static libs)" ON)
 
 add_library(pystring
     pystring.cpp
@@ -19,4 +19,7 @@ include(GNUInstallDirs)
 install(TARGETS pystring
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
 )
-
+install (FILES pystring.h
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}
+    COMPONENT developer
+)
